// @po-ui/ng-templates/esm2020/lib/components/po-page-background/po-page-background.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoDividerModule, PoFieldModule } from '@po-ui/ng-components';
import { PoPageBackgroundComponent } from './po-page-background.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-background.
 */
export class PoPageBackgroundModule {
}
PoPageBackgroundModule.ɵfac = function PoPageBackgroundModule_Factory(t) { return new (t || PoPageBackgroundModule)(); };
PoPageBackgroundModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageBackgroundModule });
PoPageBackgroundModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule, FormsModule, RouterModule, PoDividerModule, PoFieldModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageBackgroundModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, RouterModule, PoDividerModule, PoFieldModule],
                declarations: [PoPageBackgroundComponent],
                exports: [PoPageBackgroundComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageBackgroundModule, { declarations: [PoPageBackgroundComponent], imports: [CommonModule, FormsModule, RouterModule, PoDividerModule, PoFieldModule], exports: [PoPageBackgroundComponent] }); })();
