// @po-ui/ng-components/esm2020/lib/interceptors/interceptors.module.mjs
import { NgModule } from '@angular/core';
import { PoHttpInterceptorModule } from './po-http-interceptor/po-http-interceptor.module';
import { PoHttpRequestModule } from './po-http-request/po-http-request.module';
import * as i0 from "@angular/core";
export class PoInterceptorsModule {
}
PoInterceptorsModule.ɵfac = function PoInterceptorsModule_Factory(t) { return new (t || PoInterceptorsModule)(); };
PoInterceptorsModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoInterceptorsModule });
PoInterceptorsModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [], imports: [[PoHttpInterceptorModule, PoHttpRequestModule], PoHttpInterceptorModule, PoHttpRequestModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoInterceptorsModule, [{
        type: NgModule,
        args: [{
                declarations: [],
                imports: [PoHttpInterceptorModule, PoHttpRequestModule],
                exports: [PoHttpInterceptorModule, PoHttpRequestModule],
                providers: [],
                bootstrap: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoInterceptorsModule, { imports: [PoHttpInterceptorModule, PoHttpRequestModule], exports: [PoHttpInterceptorModule, PoHttpRequestModule] }); })();
