// @po-ui/ng-templates/esm2020/lib/components/po-page-dynamic-detail/po-page-dynamic-detail.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoDynamicModule, PoModalModule, PoPageModule, PoWidgetModule } from '@po-ui/ng-components';
import { PoPageDynamicDetailComponent } from './po-page-dynamic-detail.component';
import { PoPageDynamicModule } from '../../services/po-page-dynamic/po-page-dynamic.module';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-dynamic-detail.
 */
export class PoPageDynamicDetailModule {
}
PoPageDynamicDetailModule.ɵfac = function PoPageDynamicDetailModule_Factory(t) { return new (t || PoPageDynamicDetailModule)(); };
PoPageDynamicDetailModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageDynamicDetailModule });
PoPageDynamicDetailModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            FormsModule,
            RouterModule,
            PoDynamicModule,
            PoModalModule,
            PoPageModule,
            PoWidgetModule,
            PoPageDynamicModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageDynamicDetailModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    RouterModule,
                    PoDynamicModule,
                    PoModalModule,
                    PoPageModule,
                    PoWidgetModule,
                    PoPageDynamicModule
                ],
                declarations: [PoPageDynamicDetailComponent],
                exports: [PoPageDynamicDetailComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageDynamicDetailModule, { declarations: [PoPageDynamicDetailComponent], imports: [CommonModule,
        FormsModule,
        RouterModule,
        PoDynamicModule,
        PoModalModule,
        PoPageModule,
        PoWidgetModule,
        PoPageDynamicModule], exports: [PoPageDynamicDetailComponent] }); })();
