// @po-ui/ng-storage/esm2020/lib/po-storage.module.mjs
import { NgModule } from '@angular/core';
import { PoStorageService, PO_STORAGE_CONFIG_TOKEN } from './services/po-storage.service';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente PoStorage responsável por manipular o storage do browser.
 */
export class PoStorageModule {
    static forRoot(storageConfig) {
        return {
            ngModule: PoStorageModule,
            providers: [
                {
                    provide: PO_STORAGE_CONFIG_TOKEN,
                    useValue: storageConfig || PoStorageService.getDefaultConfig()
                },
                {
                    provide: PoStorageService,
                    useFactory: PoStorageService.providePoStorage,
                    deps: [PO_STORAGE_CONFIG_TOKEN]
                }
            ]
        };
    }
}
PoStorageModule.ɵfac = function PoStorageModule_Factory(t) { return new (t || PoStorageModule)(); };
PoStorageModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoStorageModule });
PoStorageModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({});
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoStorageModule, [{
        type: NgModule
    }], null, null); })();
