// @po-ui/ng-templates/esm2020/lib/po-templates.module.mjs
import { NgModule } from '@angular/core';
import { PoComponentsModule } from './components/components.module';
import { PoServicesModule } from './services/services.module';
import * as i0 from "@angular/core";
export class PoTemplatesModule {
}
PoTemplatesModule.ɵfac = function PoTemplatesModule_Factory(t) { return new (t || PoTemplatesModule)(); };
PoTemplatesModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoTemplatesModule });
PoTemplatesModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[PoComponentsModule, PoServicesModule], PoComponentsModule, PoServicesModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTemplatesModule, [{
        type: NgModule,
        args: [{
                imports: [PoComponentsModule, PoServicesModule],
                exports: [PoComponentsModule, PoServicesModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoTemplatesModule, { imports: [PoComponentsModule, PoServicesModule], exports: [PoComponentsModule, PoServicesModule] }); })();
