// @po-ui/ng-components/esm2020/lib/components/po-field/po-field-container/po-field-container.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoFieldContainerBottomComponent } from './po-field-container-bottom/po-field-container-bottom.component';
import { PoFieldContainerComponent } from './po-field-container.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-field-container`.
 */
export class PoFieldContainerModule {
}
PoFieldContainerModule.ɵfac = function PoFieldContainerModule_Factory(t) { return new (t || PoFieldContainerModule)(); };
PoFieldContainerModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoFieldContainerModule });
PoFieldContainerModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoFieldContainerModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [PoFieldContainerBottomComponent, PoFieldContainerComponent],
                declarations: [PoFieldContainerBottomComponent, PoFieldContainerComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoFieldContainerModule, { declarations: [PoFieldContainerBottomComponent, PoFieldContainerComponent], imports: [CommonModule], exports: [PoFieldContainerBottomComponent, PoFieldContainerComponent] }); })();
