// @po-ui/ng-components/esm2020/lib/components/po-breadcrumb/po-breadcrumb.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoBreadcrumbComponent } from './po-breadcrumb.component';
import { PoBreadcrumbDropdownComponent } from './po-breadcrumb-dropdown/po-breadcrumb-dropdown.component';
import { PoBreadcrumbFavoriteComponent } from './po-breadcrumb-favorite/po-breadcrumb-favorite.component';
import { PoBreadcrumbItemComponent } from './po-breadcrumb-item/po-breadcrumb-item.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-breadcrumb.
 *
 */
export class PoBreadcrumbModule {
}
PoBreadcrumbModule.ɵfac = function PoBreadcrumbModule_Factory(t) { return new (t || PoBreadcrumbModule)(); };
PoBreadcrumbModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoBreadcrumbModule });
PoBreadcrumbModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule, RouterModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoBreadcrumbModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule],
                declarations: [
                    PoBreadcrumbComponent,
                    PoBreadcrumbDropdownComponent,
                    PoBreadcrumbFavoriteComponent,
                    PoBreadcrumbItemComponent
                ],
                exports: [PoBreadcrumbComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoBreadcrumbModule, { declarations: [PoBreadcrumbComponent,
        PoBreadcrumbDropdownComponent,
        PoBreadcrumbFavoriteComponent,
        PoBreadcrumbItemComponent], imports: [CommonModule, RouterModule], exports: [PoBreadcrumbComponent] }); })();
