// @po-ui/ng-components/esm2020/lib/services/po-notification/po-notification.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoToasterComponent } from './po-toaster/po-toaster.component';
import * as i0 from "@angular/core";
export class PoNotificationModule {
}
PoNotificationModule.ɵfac = function PoNotificationModule_Factory(t) { return new (t || PoNotificationModule)(); };
PoNotificationModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoNotificationModule });
PoNotificationModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [], imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNotificationModule, [{
        type: NgModule,
        args: [{
                declarations: [PoToasterComponent],
                imports: [CommonModule],
                exports: [],
                providers: [],
                bootstrap: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoNotificationModule, { declarations: [PoToasterComponent], imports: [CommonModule] }); })();
