// @po-ui/ng-components/esm2020/lib/components/po-avatar/po-avatar.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoAvatarComponent } from './po-avatar.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-avatar.
 */
export class PoAvatarModule {
}
PoAvatarModule.ɵfac = function PoAvatarModule_Factory(t) { return new (t || PoAvatarModule)(); };
PoAvatarModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoAvatarModule });
PoAvatarModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoAvatarModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [PoAvatarComponent],
                exports: [PoAvatarComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoAvatarModule, { declarations: [PoAvatarComponent], imports: [CommonModule], exports: [PoAvatarComponent] }); })();
