// @po-ui/ng-components/esm2020/lib/components/po-popover/po-popover.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoPopoverComponent } from './po-popover.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-popover.
 */
export class PoPopoverModule {
}
PoPopoverModule.ɵfac = function PoPopoverModule_Factory(t) { return new (t || PoPopoverModule)(); };
PoPopoverModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPopoverModule });
PoPopoverModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPopoverModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [PoPopoverComponent],
                exports: [PoPopoverComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPopoverModule, { declarations: [PoPopoverComponent], imports: [CommonModule], exports: [PoPopoverComponent] }); })();
