// @po-ui/ng-components/esm2020/lib/components/po-table/po-table.component.mjs
import { Component, ContentChild, ElementRef, ViewChild, ViewChildren, ContentChildren } from '@angular/core';
import { convertToBoolean } from '../../utils/util';
import { PoDateService } from '../../services/po-date/po-date.service';
import { PoTableBaseComponent } from './po-table-base.component';
import { PoTableRowTemplateDirective } from './po-table-row-template/po-table-row-template.directive';
import { PoTableCellTemplateDirective } from './po-table-cell-template/po-table-cell-template.directive';
import { PoTableColumnTemplateDirective } from './po-table-column-template/po-table-column-template.directive';
import { PoTableRowTemplateArrowDirection } from './enums/po-table-row-template-arrow-direction.enum';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-date/po-date.service";
import * as i2 from "../../services/po-language/po-language.service";
import * as i3 from "@angular/common";
import * as i4 from "@angular/router";
import * as i5 from "./services/po-table.service";
import * as i6 from "../po-container/po-container.component";
import * as i7 from "../po-loading/po-loading.component";
import * as i8 from "./po-table-subtitle-footer/po-table-subtitle-footer.component";
import * as i9 from "../po-button/po-button.component";
import * as i10 from "../../directives/po-tooltip/po-tooltip.directive";
import * as i11 from "./po-table-column-link/po-table-column-link.component";
import * as i12 from "./po-table-column-icon/po-table-column-icon.component";
import * as i13 from "./po-table-subtitle-circle/po-table-subtitle-circle.component";
import * as i14 from "./po-table-column-label/po-table-column-label.component";
import * as i15 from "./po-table-detail/po-table-detail.component";
import * as i16 from "../po-popup/po-popup.component";
import * as i17 from "../po-icon/po-icon.component";
import * as i18 from "./po-table-column-manager/po-table-column-manager.component";
import * as i19 from "../../pipes/po-time/po-time.pipe";
const _c0 = ["noColumnsHeader"];
const _c1 = ["popup"];
const _c2 = ["tableFooter"];
const _c3 = ["tableWrapper"];
const _c4 = ["poTableTbody"];
const _c5 = ["columnManagerTarget"];
const _c6 = ["actionsIconElement"];
const _c7 = ["actionsElement"];
const _c8 = ["headersTable"];
function PoTableComponent_po_container_0_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_po_container_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "po-container", 14);
    i0.ɵɵtemplate(1, PoTableComponent_po_container_0_ng_container_1_Template, 1, 0, "ng-container", 15);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    const _r1 = i0.ɵɵreference(2);
    i0.ɵɵproperty("p-no-shadow", ctx_r0.container === "border");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r1);
} }
function PoTableComponent_ng_template_1_div_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 23);
    i0.ɵɵelement(1, "po-loading", 24);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r23 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-text", ctx_r23.literals.loadingData);
} }
function PoTableComponent_ng_template_1_div_5_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_1_div_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 25);
    i0.ɵɵelement(1, "div", 26);
    i0.ɵɵelementStart(2, "div", 27, 28);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_1_div_5_ng_container_4_Template, 1, 0, "ng-container", 15);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r25 = i0.ɵɵnextContext(2);
    const _r4 = i0.ɵɵreference(5);
    i0.ɵɵstyleProp("height", ctx_r25.heightTableContainer, "px");
    i0.ɵɵadvance(4);
    i0.ɵɵproperty("ngTemplateOutlet", _r4);
} }
function PoTableComponent_ng_template_1_div_6_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_1_div_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_1_div_6_ng_container_1_Template, 1, 0, "ng-container", 15);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    i0.ɵɵnextContext(2);
    const _r4 = i0.ɵɵreference(5);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r4);
} }
function PoTableComponent_ng_template_1_div_7_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵelement(1, "po-table-subtitle-footer", 31);
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const column_r32 = ctx.$implicit;
    const ctx_r31 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-literals", ctx_r31.literals)("p-subtitles", column_r32.subtitles);
} }
function PoTableComponent_ng_template_1_div_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 29);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_1_div_7_ng_container_1_Template, 2, 2, "ng-container", 30);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r27 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r27.subtitleColumns)("ngForTrackBy", ctx_r27.trackBy);
} }
function PoTableComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_1_div_1_Template, 2, 1, "div", 16);
    i0.ɵɵelementStart(2, "div", 17)(3, "div", 18, 19);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_1_div_5_Template, 5, 3, "div", 20);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_1_div_6_Template, 2, 1, "div", 21);
    i0.ɵɵelementEnd()()();
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_1_div_7_Template, 2, 2, "div", 22);
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-container-relative", ctx_r2.loading);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.loading);
    i0.ɵɵadvance(2);
    i0.ɵɵstyleProp("opacity", ctx_r2.tableOpacity);
    i0.ɵɵclassProp("po-table-header-fixed-columns-pixels", ctx_r2.allColumnsWidthPixels);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r2.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r2.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.hasFooter);
} }
function PoTableComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    const _r35 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 32, 33)(2, "po-button", 34);
    i0.ɵɵlistener("p-click", function PoTableComponent_div_3_Template_po_button_p_click_2_listener() { i0.ɵɵrestoreView(_r35); const ctx_r34 = i0.ɵɵnextContext(); return ctx_r34.onShowMore(); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-invisible", ctx_r3.showMore.observers.length === 0 && !ctx_r3.hasService);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("p-disabled", ctx_r3.showMoreDisabled)("p-label", ctx_r3.literals.loadMoreData)("p-loading", ctx_r3.loadingShowMore);
} }
function PoTableComponent_ng_template_4_th_3_input_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "input", 47);
} if (rf & 2) {
    const ctx_r46 = i0.ɵɵnextContext(3);
    i0.ɵɵclassProp("po-table-checkbox-checked", ctx_r46.selectAll)("po-table-checkbox-indeterminate", ctx_r46.selectAll === null);
} }
function PoTableComponent_ng_template_4_th_3_label_3_Template(rf, ctx) { if (rf & 1) {
    const _r49 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "label", 48);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_th_3_label_3_Template_label_click_0_listener() { i0.ɵɵrestoreView(_r49); const ctx_r48 = i0.ɵɵnextContext(3); return ctx_r48.selectAllRows(); });
    i0.ɵɵelementEnd();
} }
function PoTableComponent_ng_template_4_th_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 44)(1, "div");
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_4_th_3_input_2_Template, 1, 4, "input", 45);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_4_th_3_label_3_Template, 1, 0, "label", 46);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r36 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r36.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r36.hideSelectAll);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r36.hideSelectAll);
} }
function PoTableComponent_ng_template_4_th_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 49);
} }
function PoTableComponent_ng_template_4_th_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th");
} if (rf & 2) {
    const ctx_r38 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-master-detail", !ctx_r38.isSingleAction)("po-table-header-single-action", ctx_r38.isSingleAction);
} }
function PoTableComponent_ng_template_4_th_6_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_4_th_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 50, 51);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_4_th_6_ng_container_2_Template, 1, 0, "ng-container", 52);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r39 = i0.ɵɵnextContext(2);
    const _r15 = i0.ɵɵreference(17);
    const _r17 = i0.ɵɵreference(19);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r39.height)("ngIfThen", _r15)("ngIfElse", _r17);
} }
function PoTableComponent_ng_template_4_th_7_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
const _c9 = function (a0) { return { $implicit: a0 }; };
function PoTableComponent_ng_template_4_th_7_Template(rf, ctx) { if (rf & 1) {
    const _r57 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", 53, 54);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_th_7_Template_th_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r57); const column_r52 = restoredCtx.$implicit; const ctx_r56 = i0.ɵɵnextContext(2); return ctx_r56.sortColumn(column_r52); });
    i0.ɵɵelementStart(2, "div", 55);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_4_th_7_ng_container_3_Template, 1, 0, "ng-container", 56);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r52 = ctx.$implicit;
    const ctx_r40 = i0.ɵɵnextContext(2);
    const _r13 = i0.ɵɵreference(15);
    i0.ɵɵstyleProp("width", column_r52.width)("max-width", column_r52.width)("min-width", column_r52.width);
    i0.ɵɵclassProp("po-clickable", ctx_r40.sort && column_r52.sortable !== false || ctx_r40.hasService)("po-table-header-subtitle", column_r52.type === "subtitle");
    i0.ɵɵadvance(2);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r40.height)("po-table-header-flex-right", column_r52.type === "currency" || column_r52.type === "number")("po-table-header-flex-center", column_r52.type === "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r13)("ngTemplateOutletContext", i0.ɵɵpureFunction1(18, _c9, column_r52));
} }
function PoTableComponent_ng_template_4_th_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 49);
} }
function PoTableComponent_ng_template_4_th_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th");
} if (rf & 2) {
    const ctx_r42 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-single-action", ctx_r42.isSingleAction)("po-table-header-actions", !ctx_r42.isSingleAction);
} }
function PoTableComponent_ng_template_4_th_10_Template(rf, ctx) { if (rf & 1) {
    const _r61 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", null, 57)(2, "div")(3, "button", 58, 59);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_th_10_Template_button_click_3_listener() { i0.ɵɵrestoreView(_r61); const ctx_r60 = i0.ɵɵnextContext(2); return ctx_r60.onOpenColumnManager(); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const _r58 = i0.ɵɵreference(1);
    const ctx_r43 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-column-manager", !ctx_r43.isSingleAction || !ctx_r43.actionRight)("po-table-header-column-manager-border", !ctx_r43.height && ctx_r43.container)("po-table-header-single-action", ctx_r43.isSingleAction && ctx_r43.actionRight);
    i0.ɵɵadvance(2);
    i0.ɵɵstyleProp("width", ctx_r43.height && ctx_r43.visibleActions.length ? _r58.offsetWidth : undefined, "px");
    i0.ɵɵclassProp("po-table-header-column-manager-border", ctx_r43.height && ctx_r43.container)("po-table-header-column-manager-fixed-inner", ctx_r43.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-tooltip", ctx_r43.literals.columnsManager);
} }
function PoTableComponent_ng_template_4_tbody_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 60)(1, "tr", 61)(2, "td", 62)(3, "span");
    i0.ɵɵtext(4);
    i0.ɵɵelementEnd()()()();
} if (rf & 2) {
    const ctx_r44 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("colSpan", ctx_r44.columnCount);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r44.literals.noData, " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "td", 68);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_2_ng_container_1_Template, 1, 0, "ng-container", 56);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r65 = i0.ɵɵnextContext(3);
    const _r9 = i0.ɵɵreference(11);
    const _r11 = i0.ɵɵreference(13);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r65.singleSelect ? _r9 : _r11)("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c9, row_r63));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_3_ng_template_1_Template(rf, ctx) { }
const _c10 = function (a0, a1) { return { row: a0, rowIndex: a1 }; };
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_3_Template(rf, ctx) { if (rf & 1) {
    const _r81 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 69);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_3_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r81); const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r79 = i0.ɵɵnextContext(3); return ctx_r79.toggleDetail(row_r63); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_3_ng_template_1_Template, 0, 0, "ng-template", 70);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r82 = i0.ɵɵnextContext();
    const row_r63 = ctx_r82.$implicit;
    const rowIndex_r64 = ctx_r82.index;
    i0.ɵɵnextContext(3);
    const _r7 = i0.ɵɵreference(9);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r7)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c10, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_4_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_4_ng_container_12_tbody_1_4_ng_template_0_Template, 0, 0, "ng-template", 70);
} if (rf & 2) {
    const ctx_r84 = i0.ɵɵnextContext();
    const row_r63 = ctx_r84.$implicit;
    const rowIndex_r64 = ctx_r84.index;
    i0.ɵɵnextContext(3);
    const _r19 = i0.ɵɵreference(21);
    i0.ɵɵproperty("ngTemplateOutlet", _r19)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c10, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_5_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_5_Template(rf, ctx) { if (rf & 1) {
    const _r88 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 69);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_5_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r88); const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r86 = i0.ɵɵnextContext(3); return ctx_r86.toggleDetail(row_r63); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_5_ng_template_1_Template, 0, 0, "ng-template", 70);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r89 = i0.ɵɵnextContext();
    const row_r63 = ctx_r89.$implicit;
    const rowIndex_r64 = ctx_r89.index;
    i0.ɵɵnextContext(3);
    const _r7 = i0.ɵɵreference(9);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r7)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c10, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_2_ng_container_1_Template, 1, 0, "ng-container", 56);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r92 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r92.getTemplate(column_r90))("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c9, ctx_r92.getCellData(row_r63, column_r90)));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_3_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
const _c11 = function (a0, a1) { return { row: a0, column: a1 }; };
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_3_ng_container_1_Template, 1, 0, "ng-container", 56);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r93 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r93.tableCellTemplate == null ? null : ctx_r93.tableCellTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c11, row_r63, column_r90));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r94 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r94.getBooleanLabel(ctx_r94.getCellData(row_r63, column_r90), column_r90), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "currency");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r95 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind4(2, 1, ctx_r95.getCellData(row_r63, column_r90), column_r90.format, "symbol", "1.2-2"), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r96 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r96.getCellData(row_r63, column_r90), column_r90.format || "dd/MM/yyyy"), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "po_time");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r97 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r97.getCellData(row_r63, column_r90), column_r90.format || "HH:mm:ss.ffffff"), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r98 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r98.getCellData(row_r63, column_r90), column_r90.format || "dd/MM/yyyy HH:mm:ss"), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r99 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r99.formatNumber(ctx_r99.getCellData(row_r63, column_r90), column_r90.format), " ");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_po_table_column_link_10_Template(rf, ctx) { if (rf & 1) {
    const _r125 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-table-column-link", 77);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_po_table_column_link_10_Template_po_table_column_link_click_0_listener($event) { i0.ɵɵrestoreView(_r125); const column_r90 = i0.ɵɵnextContext().$implicit; const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r123 = i0.ɵɵnextContext(3); return ctx_r123.onClickLink($event, row_r63, column_r90); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r100 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-action", column_r90.action)("p-disabled", ctx_r100.checkDisabled(row_r63, column_r90))("p-link", row_r63[column_r90.link])("p-row", row_r63)("p-value", ctx_r100.getCellData(row_r63, column_r90));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_po_table_column_icon_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-table-column-icon", 78);
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r101 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-column", column_r90)("p-icons", ctx_r101.getColumnIcons(row_r63, column_r90))("p-row", row_r63);
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-subtitle-circle", 79);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r102 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-subtitle", ctx_r102.getSubtitleColumn(row_r63, column_r90));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_13_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-column-label", 80);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r103 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-value", ctx_r103.getColumnLabel(row_r63, column_r90));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r90 = i0.ɵɵnextContext().$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r104 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r104.getCellData(row_r63, column_r90));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_Template(rf, ctx) { if (rf & 1) {
    const _r138 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 71);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r138); const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r137 = i0.ɵɵnextContext(3); return ctx_r137.selectable ? ctx_r137.selectRow(row_r63) : "javascript:;"; });
    i0.ɵɵelementStart(1, "div", 72);
    i0.ɵɵlistener("mouseenter", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_Template_div_mouseenter_1_listener($event) { const restoredCtx = i0.ɵɵrestoreView(_r138); const column_r90 = restoredCtx.$implicit; const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r140 = i0.ɵɵnextContext(3); return ctx_r140.tooltipMouseEnter($event, column_r90, row_r63); })("mouseleave", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_Template_div_mouseleave_1_listener() { i0.ɵɵrestoreView(_r138); const ctx_r142 = i0.ɵɵnextContext(4); return ctx_r142.tooltipMouseLeave(); });
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_2_Template, 2, 4, "span", 73);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_3_Template, 2, 5, "span", 73);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_4_Template, 2, 1, "span", 73);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_5_Template, 3, 6, "span", 73);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_6_Template, 3, 4, "span", 73);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_7_Template, 3, 4, "span", 73);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_8_Template, 3, 4, "span", 73);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_9_Template, 2, 1, "span", 73);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_po_table_column_link_10_Template, 1, 5, "po-table-column-link", 74);
    i0.ɵɵtemplate(11, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_po_table_column_icon_11_Template, 1, 3, "po-table-column-icon", 75);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_12_Template, 2, 1, "span", 73);
    i0.ɵɵtemplate(13, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_13_Template, 2, 1, "span", 73);
    i0.ɵɵtemplate(14, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_span_14_Template, 2, 1, "span", 76);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r90 = ctx.$implicit;
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r69 = i0.ɵɵnextContext(3);
    i0.ɵɵstyleProp("width", column_r90.width)("max-width", column_r90.width)("min-width", column_r90.width);
    i0.ɵɵclassProp("po-table-column", column_r90.type !== "icon")("po-table-column-right", column_r90.type === "currency" || column_r90.type === "number")("po-table-column-center", column_r90.type === "subtitle")("po-table-column-icons", column_r90.type === "icon");
    i0.ɵɵproperty("ngClass", ctx_r69.getClassColor(row_r63, column_r90));
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-body-ellipsis", ctx_r69.hideTextOverflow);
    i0.ɵɵproperty("ngSwitch", column_r90.type)("p-tooltip", ctx_r69.tooltipText);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "columnTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "cellTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "boolean");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "currency");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "date");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "time");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "dateTime");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "number");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "link");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "icon");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "label");
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_7_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_7_Template(rf, ctx) { if (rf & 1) {
    const _r147 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 69);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_7_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r147); const row_r63 = i0.ɵɵnextContext().$implicit; const ctx_r145 = i0.ɵɵnextContext(3); return ctx_r145.toggleDetail(row_r63); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_7_ng_template_1_Template, 0, 0, "ng-template", 70);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r148 = i0.ɵɵnextContext();
    const row_r63 = ctx_r148.$implicit;
    const rowIndex_r64 = ctx_r148.index;
    i0.ɵɵnextContext(3);
    const _r7 = i0.ɵɵreference(9);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r7)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c10, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_8_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_4_ng_container_12_tbody_1_8_ng_template_0_Template, 0, 0, "ng-template", 70);
} if (rf & 2) {
    const ctx_r150 = i0.ɵɵnextContext();
    const row_r63 = ctx_r150.$implicit;
    const rowIndex_r64 = ctx_r150.index;
    i0.ɵɵnextContext(3);
    const _r19 = i0.ɵɵreference(21);
    i0.ɵɵproperty("ngTemplateOutlet", _r19)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c10, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_ng_container_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵelement(1, "td", 81);
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_10_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "td", 81);
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_11_ng_template_2_Template(rf, ctx) { }
const _c12 = function (a0, a1) { return { $implicit: a0, rowIndex: a1 }; };
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tr")(1, "td", 82);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_11_ng_template_2_Template, 0, 0, "ng-template", 70);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r152 = i0.ɵɵnextContext();
    const row_r63 = ctx_r152.$implicit;
    const rowIndex_r64 = ctx_r152.index;
    const ctx_r74 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r74.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r74.tableRowTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(3, _c12, row_r63, rowIndex_r64));
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_12_Template(rf, ctx) { if (rf & 1) {
    const _r154 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "tr")(1, "td", 83)(2, "po-table-detail", 84);
    i0.ɵɵlistener("p-select-row", function PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_12_Template_po_table_detail_p_select_row_2_listener($event) { i0.ɵɵrestoreView(_r154); const ctx_r153 = i0.ɵɵnextContext(4); return ctx_r153.selectDetailRow($event); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const row_r63 = i0.ɵɵnextContext().$implicit;
    const ctx_r75 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r75.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-selectable", ctx_r75.selectable && !ctx_r75.detailHideSelect)("p-detail", ctx_r75.columnMasterDetail.detail)("p-items", row_r63[ctx_r75.nameColumnDetail]);
} }
function PoTableComponent_ng_template_4_ng_container_12_tbody_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 60)(1, "tr", 61);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_2_Template, 2, 4, "td", 64);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_3_Template, 2, 5, "td", 65);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_4_ng_container_12_tbody_1_4_Template, 1, 5, null, 21);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_5_Template, 2, 5, "td", 65);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_6_Template, 15, 31, "td", 66);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_7_Template, 2, 5, "td", 65);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_4_ng_container_12_tbody_1_8_Template, 1, 5, null, 21);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_4_ng_container_12_tbody_1_ng_container_9_Template, 2, 0, "ng-container", 21);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_4_ng_container_12_tbody_1_td_10_Template, 1, 0, "td", 67);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(11, PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_11_Template, 3, 6, "tr", 21);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_4_ng_container_12_tbody_1_tr_12_Template, 3, 4, "tr", 21);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r63 = ctx.$implicit;
    const rowIndex_r64 = ctx.index;
    const ctx_r62 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-row-active", row_r63.$selected || row_r63.$selected === null && ctx_r62.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.columnMasterDetail && !ctx_r62.hideDetail && !ctx_r62.hasRowTemplate);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r62.actionRight && (ctx_r62.visibleActions.length > 1 || ctx_r62.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.hasRowTemplate && !ctx_r62.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r62.mainColumns)("ngForTrackBy", ctx_r62.trackBy);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.actionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r62.actionRight && (ctx_r62.visibleActions.length > 1 || ctx_r62.isSingleAction) && !ctx_r62.hideColumnsManager);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r62.hasVisibleActions && !ctx_r62.hideColumnsManager && !ctx_r62.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.hasMainColumns && ctx_r62.hasRowTemplate && row_r63.$showDetail && ctx_r62.isShowRowTemplate(row_r63, rowIndex_r64));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r62.hasMainColumns && ctx_r62.isShowMasterDetail(row_r63));
} }
function PoTableComponent_ng_template_4_ng_container_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_4_ng_container_12_tbody_1_Template, 13, 14, "tbody", 63);
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r45 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r45.items)("ngForTrackBy", ctx_r45.trackBy);
} }
function PoTableComponent_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "table", 35)(1, "thead")(2, "tr");
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_4_th_3_Template, 4, 4, "th", 36);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_4_th_4_Template, 1, 0, "th", 37);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_4_th_5_Template, 1, 4, "th", 38);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_4_th_6_Template, 3, 3, "th", 39);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_4_th_7_Template, 4, 20, "th", 40);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_4_th_8_Template, 1, 0, "th", 37);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_4_th_9_Template, 1, 4, "th", 41);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_4_th_10_Template, 5, 13, "th", 42);
    i0.ɵɵelementEnd()();
    i0.ɵɵtemplate(11, PoTableComponent_ng_template_4_tbody_11_Template, 5, 2, "tbody", 43);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_4_ng_container_12_Template, 2, 2, "ng-container", 21);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-striped", ctx_r5.striped)("po-table-layout-fixed", ctx_r5.hideTextOverflow);
    i0.ɵɵadvance(2);
    i0.ɵɵclassProp("po-table-header", !ctx_r5.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasSelectableColumn);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", (ctx_r5.hasMasterDetailColumn || ctx_r5.hasRowTemplate) && !ctx_r5.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r5.actionRight && (ctx_r5.visibleActions.length > 1 || ctx_r5.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r5.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r5.mainColumns)("ngForTrackBy", ctx_r5.trackBy);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasRowTemplateWithArrowDirectionRight && (ctx_r5.hasVisibleActions || ctx_r5.hideColumnsManager));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasVisibleActions && ctx_r5.hideColumnsManager && ctx_r5.actionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasValidColumns && !ctx_r5.hideColumnsManager);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r5.hasItems || !ctx_r5.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasMainColumns);
} }
function PoTableComponent_ng_template_8_span_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "span", 86);
} if (rf & 2) {
    const row_r156 = i0.ɵɵnextContext().row;
    i0.ɵɵclassProp("po-icon-arrow-up", row_r156.$showDetail)("po-icon-arrow-down", !row_r156.$showDetail);
} }
function PoTableComponent_ng_template_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_8_span_0_Template, 1, 4, "span", 85);
} if (rf & 2) {
    const row_r156 = ctx.row;
    const rowIndex_r157 = ctx.rowIndex;
    const ctx_r8 = i0.ɵɵnextContext();
    i0.ɵɵproperty("ngIf", ctx_r8.containsMasterDetail(row_r156) && !ctx_r8.hasRowTemplate || ctx_r8.isShowRowTemplate(row_r156, rowIndex_r157) && ctx_r8.hasRowTemplate);
} }
function PoTableComponent_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    const _r162 = i0.ɵɵgetCurrentView();
    i0.ɵɵelement(0, "input", 87);
    i0.ɵɵelementStart(1, "label", 88);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_Template_label_click_1_listener() { const restoredCtx = i0.ɵɵrestoreView(_r162); const row_r160 = restoredCtx.$implicit; const ctx_r161 = i0.ɵɵnextContext(); return ctx_r161.selectable ? ctx_r161.selectRow(row_r160) : "javascript:;"; });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r160 = ctx.$implicit;
    i0.ɵɵclassProp("po-table-radio-checked", row_r160.$selected);
} }
function PoTableComponent_ng_template_12_Template(rf, ctx) { if (rf & 1) {
    const _r165 = i0.ɵɵgetCurrentView();
    i0.ɵɵelement(0, "input", 47);
    i0.ɵɵelementStart(1, "label", 48);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_12_Template_label_click_1_listener() { const restoredCtx = i0.ɵɵrestoreView(_r165); const row_r163 = restoredCtx.$implicit; const ctx_r164 = i0.ɵɵnextContext(); return ctx_r164.selectable ? ctx_r164.selectRow(row_r163) : "javascript:;"; });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r163 = ctx.$implicit;
    i0.ɵɵclassProp("po-table-checkbox-checked", row_r163.$selected);
} }
function PoTableComponent_ng_template_14_span_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "span");
} if (rf & 2) {
    const column_r166 = i0.ɵɵnextContext().$implicit;
    const ctx_r168 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-header-icon-unselected", (ctx_r168.sortedColumn == null ? null : ctx_r168.sortedColumn.property) !== column_r166)("po-table-header-icon-descending", (ctx_r168.sortedColumn == null ? null : ctx_r168.sortedColumn.property) === column_r166 && ctx_r168.sortedColumn.ascending)("po-table-header-icon-ascending", (ctx_r168.sortedColumn == null ? null : ctx_r168.sortedColumn.property) === column_r166 && !ctx_r168.sortedColumn.ascending);
} }
function PoTableComponent_ng_template_14_Template(rf, ctx) { if (rf & 1) {
    const _r171 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "span", 89, 90);
    i0.ɵɵlistener("mouseenter", function PoTableComponent_ng_template_14_Template_span_mouseenter_0_listener($event) { i0.ɵɵrestoreView(_r171); const ctx_r170 = i0.ɵɵnextContext(); return ctx_r170.tooltipMouseEnter($event); })("mouseleave", function PoTableComponent_ng_template_14_Template_span_mouseleave_0_listener() { i0.ɵɵrestoreView(_r171); const ctx_r172 = i0.ɵɵnextContext(); return ctx_r172.tooltipMouseLeave(); });
    i0.ɵɵtext(2);
    i0.ɵɵpipe(3, "titlecase");
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_14_span_4_Template, 1, 6, "span", 91);
} if (rf & 2) {
    const column_r166 = ctx.$implicit;
    const ctx_r14 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-tooltip", ctx_r14.tooltipText);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", column_r166.label || i0.ɵɵpipeBind1(3, 3, column_r166.property), " ");
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r14.sort && column_r166.sortable !== false);
} }
function PoTableComponent_ng_template_16_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 92);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r16 = i0.ɵɵnextContext();
    i0.ɵɵstyleProp("width", ctx_r16.noColumnsHeader == null ? null : ctx_r16.noColumnsHeader.nativeElement.offsetWidth, "px");
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r16.hasValidColumns ? ctx_r16.literals.noVisibleColumn : ctx_r16.literals.noColumns, " ");
} }
function PoTableComponent_ng_template_18_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtext(0);
} if (rf & 2) {
    const ctx_r18 = i0.ɵɵnextContext();
    i0.ɵɵtextInterpolate1(" ", ctx_r18.hasValidColumns ? ctx_r18.literals.noVisibleColumn : ctx_r18.literals.noColumns, "\n");
} }
function PoTableComponent_ng_template_20_td_0_div_1_po_icon_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 99);
} if (rf & 2) {
    const ctx_r178 = i0.ɵɵnextContext(4);
    i0.ɵɵproperty("p-icon", ctx_r178.firstAction.icon);
} }
function PoTableComponent_ng_template_20_td_0_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r181 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 97);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_20_td_0_div_1_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r181); const row_r173 = i0.ɵɵnextContext(2).row; const ctx_r179 = i0.ɵɵnextContext(); return ctx_r179.executeTableAction(row_r173, ctx_r179.firstAction); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_20_td_0_div_1_po_icon_1_Template, 1, 1, "po-icon", 98);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r173 = i0.ɵɵnextContext(2).row;
    const ctx_r177 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-action-disabled", ctx_r177.firstAction.disabled ? ctx_r177.validateTableAction(row_r173, ctx_r177.firstAction) : false);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r177.firstAction.icon);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r177.firstAction.label, " ");
} }
function PoTableComponent_ng_template_20_td_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_20_td_0_div_1_Template, 3, 4, "div", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r175 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r175.firstAction.visible !== false);
} }
function PoTableComponent_ng_template_20_td_1_Template(rf, ctx) { if (rf & 1) {
    const _r186 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 100)(1, "span", 101, 102);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_20_td_1_Template_span_click_1_listener() { i0.ɵɵrestoreView(_r186); const _r183 = i0.ɵɵreference(2); const row_r173 = i0.ɵɵnextContext().row; const ctx_r184 = i0.ɵɵnextContext(); return ctx_r184.togglePopup(row_r173, _r183); });
    i0.ɵɵelementEnd()();
} }
function PoTableComponent_ng_template_20_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_20_td_0_Template, 2, 1, "td", 93);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_20_td_1_Template, 3, 0, "td", 94);
} if (rf & 2) {
    const ctx_r20 = i0.ɵɵnextContext();
    i0.ɵɵproperty("ngIf", ctx_r20.isSingleAction);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r20.visibleActions.length > 1);
} }
function PoTableComponent_po_table_column_manager_22_Template(rf, ctx) { if (rf & 1) {
    const _r188 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-table-column-manager", 103);
    i0.ɵɵlistener("p-visible-columns-change", function PoTableComponent_po_table_column_manager_22_Template_po_table_column_manager_p_visible_columns_change_0_listener($event) { i0.ɵɵrestoreView(_r188); const ctx_r187 = i0.ɵɵnextContext(); return ctx_r187.onVisibleColumnsChange($event); })("p-change-visible-columns", function PoTableComponent_po_table_column_manager_22_Template_po_table_column_manager_p_change_visible_columns_0_listener($event) { i0.ɵɵrestoreView(_r188); const ctx_r189 = i0.ɵɵnextContext(); return ctx_r189.onChangeVisibleColumns($event); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r21 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-columns", ctx_r21.columns)("p-max-columns", ctx_r21.maxColumns)("p-target", ctx_r21.columnManagerTarget)("p-last-visible-columns-selected", ctx_r21.lastVisibleColumnsSelected);
} }
/**
 * @docsExtends PoTableBaseComponent
 *
 * @example
 *
 * <example name="po-table-basic" title="PO Table Basic">
 *  <file name="sample-po-table-basic/sample-po-table-basic.component.ts"> </file>
 *  <file name="sample-po-table-basic/sample-po-table-basic.component.html"> </file>
 * </example>
 *
 * <example name="po-table-labs" title="PO Table Labs">
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.html"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.e2e-spec.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.po.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-with-api" title="PO Table using API">
 *  <file name="sample-po-table-with-api/sample-po-table-with-api.component.ts"> </file>
 *  <file name="sample-po-table-with-api/sample-po-table-with-api.component.html"> </file>
 * </example>
 *
 * <example name="po-table-transport" title="PO Table - Transport">
 *  <file name="sample-po-table-transport/sample-po-table-transport.component.ts"> </file>
 *  <file name="sample-po-table-transport/sample-po-table-transport.component.html"> </file>
 *  <file name="sample-po-table-transport/sample-po-table-transport.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-airfare" title="PO Table - Airfare">
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.component.ts"> </file>
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.component.html"> </file>
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-components" title="PO Table - Po Field Components">
 *  <file name="sample-po-table-components/sample-po-table-components.component.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.enum.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.component.html"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.service.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.component.css"> </file>
 * </example>
 *
 */
export class PoTableComponent extends PoTableBaseComponent {
    constructor(poDate, differs, renderer, poLanguageService, changeDetector, decimalPipe, router, defaultService) {
        super(poDate, poLanguageService, defaultService);
        this.changeDetector = changeDetector;
        this.decimalPipe = decimalPipe;
        this.router = router;
        this.defaultService = defaultService;
        this.tableOpacity = 0;
        this.initialized = false;
        this.visibleElement = false;
        this.differ = differs.find([]).create(null);
        // TODO: #5550 ao remover este listener, no portal, quando as colunas forem fixas não sofrem
        // alteração de largura, pois o ngDoCheck não é executado.
        this.clickListener = renderer.listen('document', 'click', () => { });
        this.resizeListener = renderer.listen('window', 'resize', (event) => {
            this.debounceResize();
        });
    }
    set columnManagerTarget(value) {
        this._columnManagerTarget = value;
        this.changeDetector.detectChanges();
    }
    get columnManagerTarget() {
        return this._columnManagerTarget;
    }
    get hasRowTemplateWithArrowDirectionRight() {
        return this.tableRowTemplate?.tableRowTemplateArrowDirection === PoTableRowTemplateArrowDirection.Right;
    }
    get columnCount() {
        const columnCount = this.mainColumns.length +
            (this.actions.length > 0 ? 1 : 0) +
            (this.selectable ? 1 : 0) +
            (!this.hideDetail && this.columnMasterDetail !== undefined ? 1 : 0);
        return columnCount || 1;
    }
    get columnCountForMasterDetail() {
        // caso tiver ações será utilizado a sua coluna para exibir o columnManager
        const columnManager = this.actions.length ? 0 : 1;
        return this.mainColumns.length + 1 + (this.actions.length > 0 ? 1 : 0) + (this.selectable ? 1 : 0) + columnManager;
    }
    get detailHideSelect() {
        const masterDetail = this.columnMasterDetail;
        return masterDetail && masterDetail.detail ? masterDetail.detail.hideSelect : false;
    }
    get hasVisibleActions() {
        return !!this.visibleActions.length;
    }
    get firstAction() {
        return this.visibleActions && this.visibleActions[0];
    }
    get hasFooter() {
        return this.hasItems && this.hasVisibleSubtitleColumns;
    }
    get hasMasterDetailColumn() {
        return (this.hasMainColumns && this.hasItems && !this.hideDetail && !!(this.columnMasterDetail || this.hasRowTemplate));
    }
    get hasRowTemplate() {
        return !!this.tableRowTemplate;
    }
    get hasSelectableColumn() {
        return this.selectable && this.hasItems && this.hasMainColumns;
    }
    get hasValidColumns() {
        return !!this.validColumns.length;
    }
    get hasVisibleSubtitleColumns() {
        return this.subtitleColumns.some(column => column.visible !== false);
    }
    get isSingleAction() {
        return this.visibleActions.length === 1;
    }
    get visibleActions() {
        return this.actions && this.actions.filter(action => action && action.visible !== false);
    }
    ngAfterViewInit() {
        this.initialized = true;
    }
    showMoreInfiniteScroll({ target }) {
        const scrollPosition = target.offsetHeight + target.scrollTop;
        if (!this.showMoreDisabled && scrollPosition >= target.scrollHeight * (this.infiniteScrollDistance / 110)) {
            this.onShowMore();
        }
    }
    ngDoCheck() {
        this.checkChangesItems();
        this.verifyCalculateHeightTableContainer();
        // Permite que os cabeçalhos sejam calculados na primeira vez que o componente torna-se visível,
        // evitando com isso, problemas com Tabs ou Divs que iniciem escondidas.
        if (this.tableWrapperElement?.nativeElement.offsetWidth && !this.visibleElement && this.initialized) {
            this.debounceResize();
            this.checkInfiniteScroll();
            this.visibleElement = true;
        }
    }
    ngOnDestroy() {
        this.removeListeners();
    }
    /**
     * Método responsável por realizar busca no serviço de dados podendo informar filtros e com o retorno, atualiza a tabela.
     *
     * Caso não seja informado parâmetro, nada será adicionado ao GET, conforme abaixo:
     * ```
     * url + ?page=1&pageSize=10
     * ```
     * > Obs: os parâmetros `page` e `pageSize` sempre serão chamados independente de ser enviados outros parâmetros.
     *
     * Caso sejam informados os parâmetros `{ name: 'JOHN', age: '23' }`, todos serão adicionados ao GET, conforme abaixo:
     * ```
     * url + ?page=1&pageSize=10&name=JOHN&age=23
     * ```
     *
     * @param { { key: value } } queryParams Formato do objeto a ser enviado.
     * > Pode ser utilizada qualquer string como key, e qualquer string ou number como value.
     */
    applyFilters(queryParams) {
        this.page = 1;
        this.initializeData(queryParams);
    }
    /**
     * Método que colapsa uma linha com detalhe quando executada.
     *
     * @param { number } rowIndex Índice da linha que será colapsada.
     * > Ao reordenar os dados da tabela, o valor contido neste índice será alterado conforme a ordenação.
     */
    collapse(rowIndex) {
        this.setShowDetail(rowIndex, false);
    }
    /**
     * Método que expande uma linha com detalhe quando executada.
     *
     * @param { number } rowIndex Índice da linha que será expandida.
     * > Ao reordenar os dados da tabela, o valor contido neste índice será alterado conforme a ordenação.
     */
    expand(rowIndex) {
        this.setShowDetail(rowIndex, true);
    }
    /**
     * Retorna as linhas do `po-table` que estão selecionadas.
     */
    getSelectedRows() {
        return this.items.filter(item => item.$selected);
    }
    /**
     * Retorna as linhas do `po-table` que não estão selecionadas.
     */
    getUnselectedRows() {
        return this.items.filter(item => !item.$selected);
    }
    /**
     * Desmarca as linhas que estão selecionadas.
     */
    unselectRows() {
        const columnDetail = this.nameColumnDetail;
        this.items.forEach(item => {
            const detailItems = columnDetail ? item[columnDetail] : null;
            if (Array.isArray(detailItems)) {
                detailItems.forEach(detailItem => {
                    detailItem.$selected = false;
                });
            }
            item.$selected = false;
        });
        this.selectAll = false;
    }
    checkDisabled(row, column) {
        return column.disabled ? column.disabled(row) : false;
    }
    containsMasterDetail(row) {
        return row[this.nameColumnDetail] && row[this.nameColumnDetail].length;
    }
    executeTableAction(row, tableAction) {
        if (!row.disabled && !this.validateTableAction(row, tableAction)) {
            tableAction.action(row);
            this.toggleRowAction(row);
        }
    }
    /**
     * Desmarca uma linha que está selecionada.
     */
    unselectRowItem(itemfn) {
        this.toggleSelect(itemfn, false);
        if (this.items.every(item => !item.$selected)) {
            this.selectAll = false;
        }
        else {
            this.selectAll = null;
        }
    }
    /**
     * Seleciona uma linha do 'po-table'.
     */
    selectRowItem(itemfn) {
        this.toggleSelect(itemfn, true);
        if (this.items.every(item => item.$selected)) {
            this.selectAll = true;
        }
        else {
            this.selectAll = null;
        }
    }
    formatNumber(value, format) {
        if (!format) {
            return value;
        }
        return this.decimalPipe.transform(value, format);
    }
    getCellData(row, column) {
        const arrayProperty = column.property.split('.');
        if (arrayProperty.length > 1) {
            const nestedProperties = arrayProperty;
            let value = row;
            for (const property of nestedProperties) {
                value = value[property] || value[property] === 0 ? value[property] : '';
            }
            return value;
        }
        else {
            return row[column.property];
        }
    }
    getBooleanLabel(rowValue, columnBoolean) {
        if (rowValue || rowValue === false || rowValue === 0) {
            rowValue = convertToBoolean(rowValue);
            if (columnBoolean.boolean) {
                return rowValue ? columnBoolean.boolean.trueLabel || 'Sim' : columnBoolean.boolean.falseLabel || 'Não';
            }
            else {
                return rowValue ? 'Sim' : 'Não';
            }
        }
        return rowValue;
    }
    getColumnIcons(row, column) {
        const rowIcons = this.getCellData(row, column);
        if (column.icons) {
            if (Array.isArray(rowIcons)) {
                return this.mergeCustomIcons(rowIcons, column.icons);
            }
            else {
                return this.findCustomIcon(rowIcons, column);
            }
        }
        return rowIcons;
    }
    getColumnLabel(row, columnLabel) {
        return columnLabel.labels.find(labelItem => this.getCellData(row, columnLabel) === labelItem.value);
    }
    getSubtitleColumn(row, subtitleColumn) {
        return subtitleColumn.subtitles.find(subtitleItem => this.getCellData(row, subtitleColumn) === subtitleItem.value);
    }
    isShowMasterDetail(row) {
        return (!this.hideDetail &&
            this.nameColumnDetail &&
            row.$showDetail &&
            this.containsMasterDetail(row) &&
            !this.hasRowTemplate);
    }
    isShowRowTemplate(row, index) {
        if (this.tableRowTemplate && this.tableRowTemplate.poTableRowTemplateShow) {
            return this.tableRowTemplate.poTableRowTemplateShow(row, index);
        }
        return true;
    }
    onClickLink(event, row, column) {
        if (!this.checkDisabled(row, column)) {
            event.stopPropagation();
        }
    }
    onChangeVisibleColumns(columns) {
        this.changeVisibleColumns.emit(columns);
    }
    onVisibleColumnsChange(columns) {
        this.columns = columns;
        this.changeDetector.detectChanges();
    }
    tooltipMouseEnter(event, column, row) {
        this.tooltipText = undefined;
        if (this.hideTextOverflow && event.target.offsetWidth < event.target.scrollWidth && event.target.innerText.trim()) {
            return (this.tooltipText = event.target.innerText);
        }
        if (column) {
            this.checkingIfColumnHasTooltip(column, row);
        }
    }
    tooltipMouseLeave() {
        this.tooltipText = undefined;
    }
    togglePopup(row, targetRef) {
        this.popupTarget = targetRef;
        this.changeDetector.detectChanges();
        this.poPopupComponent.toggle(row);
    }
    trackBy(index) {
        return index;
    }
    validateTableAction(row, tableAction) {
        if (typeof tableAction.disabled === 'function') {
            return tableAction.disabled(row);
        }
        else {
            return tableAction.disabled;
        }
    }
    onOpenColumnManager() {
        this.lastVisibleColumnsSelected = [...this.columns];
    }
    /**
     * Método que remove um item da tabela.
     *
     * @param { number | { key: value } } item Índice da linha ou o item que será removido.
     * > Ao remover o item, a linha que o representa será excluída da tabela.
     */
    removeItem(item) {
        if (item instanceof Object) {
            this.items = this.items.filter(filterItem => filterItem !== item);
        }
        else if (typeof item === 'number') {
            const index = item;
            this.items.splice(index, 1);
        }
    }
    /**
     * Método que atualiza um item da tabela.
     *
     * @param { number | { key: value } } item Índice da linha ou o item que será atualizado.
     * @param { { key: value } } updatedItem Item que foi atualizado.
     * > Ao atualizar o item, a informação será alterada na tabela.
     */
    updateItem(item, updatedItem) {
        if (typeof item === 'number') {
            this.items.splice(item, 1, updatedItem);
        }
        else {
            const index = this.items.findIndex(indexItem => indexItem === item);
            this.items.splice(index, 1, updatedItem);
        }
    }
    getTemplate(column) {
        const template = this.tableColumnTemplates.find(tableColumnTemplate => tableColumnTemplate.targetProperty === column.property);
        if (!template) {
            console.warn(`Não foi possível encontrar o template para a coluna: ${column.property}, por gentileza informe a propriedade [p-property]`);
            return null;
        }
        return template.templateRef;
    }
    calculateHeightTableContainer(height) {
        const value = parseFloat(height);
        this.heightTableContainer = value ? value - this.getHeightTableFooter() : undefined;
        this.setTableOpacity(1);
        this.changeDetector.detectChanges();
    }
    calculateWidthHeaders() {
        setTimeout(() => {
            if (this.height) {
                this.headersTable.forEach(header => {
                    const divHeader = header.nativeElement.querySelector('.po-table-header-fixed-inner');
                    if (divHeader) {
                        divHeader.style.width = `${header.nativeElement.offsetWidth}px`;
                    }
                });
            }
        });
    }
    checkInfiniteScroll() {
        if (this.hasInfiniteScroll()) {
            if (this.poTableTbody.nativeElement.scrollHeight > this.height) {
                this.includeInfiniteScroll();
            }
            else {
                this.infiniteScroll = false;
            }
        }
        this.changeDetector.detectChanges();
    }
    checkChangesItems() {
        const changesItems = this.differ.diff(this.items);
        if (changesItems && this.selectAll) {
            this.selectAll = null;
        }
        if (changesItems && !this.hasColumns && this.hasItems) {
            this.columns = this.getDefaultColumns(this.items[0]);
        }
    }
    checkingIfColumnHasTooltip(column, row) {
        if (column.type === 'link' && column.tooltip && !this.checkDisabled(row, column)) {
            return (this.tooltipText = column.tooltip);
        }
        if (column.type === 'label') {
            const columnLabel = this.getColumnLabel(row, column);
            return (this.tooltipText = columnLabel?.tooltip);
        }
    }
    debounceResize() {
        clearTimeout(this.timeoutResize);
        this.timeoutResize = setTimeout(() => {
            this.calculateWidthHeaders();
            // show the table
            this.setTableOpacity(1);
        });
    }
    findCustomIcon(rowIcons, column) {
        const customIcon = column.icons.find(icon => rowIcons === icon.value);
        return customIcon ? [customIcon] : undefined;
    }
    getHeightTableFooter() {
        return this.tableFooterElement ? this.tableFooterElement.nativeElement.offsetHeight : 0;
    }
    hasInfiniteScroll() {
        return (this.infiniteScroll &&
            this.hasItems &&
            !this.subscriptionScrollEvent &&
            this.height &&
            this.poTableTbody.nativeElement.scrollHeight);
    }
    includeInfiniteScroll() {
        this.scrollEvent$ = this.defaultService.scrollListener(this.poTableTbody.nativeElement);
        this.subscriptionScrollEvent = this.scrollEvent$.subscribe(event => this.showMoreInfiniteScroll(event));
    }
    mergeCustomIcons(rowIcons, customIcons) {
        const mergedIcons = [];
        rowIcons.forEach(columnValue => {
            const foundCustomIcon = customIcons.find(customIcon => columnValue === customIcon.icon || columnValue === customIcon.value);
            foundCustomIcon ? mergedIcons.push(foundCustomIcon) : mergedIcons.push(columnValue);
        });
        return mergedIcons;
    }
    removeListeners() {
        if (this.resizeListener) {
            this.resizeListener();
        }
        if (this.clickListener) {
            this.clickListener();
        }
        if (this.subscriptionScrollEvent) {
            this.subscriptionScrollEvent.unsubscribe();
        }
    }
    setTableOpacity(value) {
        this.tableOpacity = value;
    }
    verifyChangeHeightInFooter() {
        return this.footerHeight !== this.getHeightTableFooter();
    }
    verifyCalculateHeightTableContainer() {
        if (this.height && this.verifyChangeHeightInFooter()) {
            this.footerHeight = this.getHeightTableFooter();
            this.calculateHeightTableContainer(this.height);
        }
    }
    toggleSelect(compare, selectValue) {
        if (typeof compare !== 'function') {
            this.items.forEach(item => {
                if (item === compare) {
                    item.$selected = selectValue;
                }
            });
        }
        else {
            this.items.forEach(item => {
                if (compare(item)) {
                    item.$selected = selectValue;
                }
            });
        }
    }
}
PoTableComponent.ɵfac = function PoTableComponent_Factory(t) { return new (t || PoTableComponent)(i0.ɵɵdirectiveInject(i1.PoDateService), i0.ɵɵdirectiveInject(i0.IterableDiffers), i0.ɵɵdirectiveInject(i0.Renderer2), i0.ɵɵdirectiveInject(i2.PoLanguageService), i0.ɵɵdirectiveInject(i0.ChangeDetectorRef), i0.ɵɵdirectiveInject(i3.DecimalPipe), i0.ɵɵdirectiveInject(i4.Router), i0.ɵɵdirectiveInject(i5.PoTableService)); };
PoTableComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTableComponent, selectors: [["po-table"]], contentQueries: function PoTableComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        i0.ɵɵcontentQuery(dirIndex, PoTableRowTemplateDirective, 7);
        i0.ɵɵcontentQuery(dirIndex, PoTableCellTemplateDirective, 5);
        i0.ɵɵcontentQuery(dirIndex, PoTableColumnTemplateDirective, 4);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableRowTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableCellTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableColumnTemplates = _t);
    } }, viewQuery: function PoTableComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 5, ElementRef);
        i0.ɵɵviewQuery(_c1, 5);
        i0.ɵɵviewQuery(_c2, 5, ElementRef);
        i0.ɵɵviewQuery(_c3, 5, ElementRef);
        i0.ɵɵviewQuery(_c4, 5, ElementRef);
        i0.ɵɵviewQuery(_c5, 5);
        i0.ɵɵviewQuery(_c6, 5, ElementRef);
        i0.ɵɵviewQuery(_c7, 5, ElementRef);
        i0.ɵɵviewQuery(_c8, 5);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.noColumnsHeader = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poPopupComponent = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableFooterElement = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableWrapperElement = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poTableTbody = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnManagerTarget = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.actionsIconElement = _t);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.actionsElement = _t);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.headersTable = _t);
    } }, features: [i0.ɵɵProvidersFeature([PoDateService]), i0.ɵɵInheritDefinitionFeature], decls: 23, vars: 6, consts: [["p-no-padding", "", 3, "p-no-shadow", 4, "ngIf", "ngIfElse"], ["tableContainerTemplate", ""], ["class", "po-row po-table-footer-show-more", 3, "po-invisible", 4, "ngIf"], ["tableTemplate", ""], [3, "p-actions", "p-target"], ["popup", ""], ["poTableColumnDetail", ""], ["inputRadio", ""], ["inputCheckbox", ""], ["contentHeaderTemplate", ""], ["noColumnsWithHeight", ""], ["noColumnsWithoutHeight", ""], ["ActionsColumnTemplate", ""], [3, "p-columns", "p-max-columns", "p-target", "p-last-visible-columns-selected", "p-visible-columns-change", "p-change-visible-columns", 4, "ngIf"], ["p-no-padding", "", 3, "p-no-shadow"], [4, "ngTemplateOutlet"], ["class", "po-table-overlay", 4, "ngIf"], [1, "po-table-main-container"], [1, "po-table-wrapper"], ["tableWrapper", ""], ["class", "po-table-container", 3, "height", 4, "ngIf"], [4, "ngIf"], ["class", "po-table-footer", 4, "ngIf"], [1, "po-table-overlay"], [1, "po-table-overlay-content", 3, "p-text"], [1, "po-table-container"], [1, "po-table-header-fixed", "po-table-header"], [1, "po-table-container-fixed-inner"], ["poTableTbody", ""], [1, "po-table-footer"], [4, "ngFor", "ngForOf", "ngForTrackBy"], [3, "p-literals", "p-subtitles"], [1, "po-row", "po-table-footer-show-more"], ["tableFooter", ""], [1, "po-offset-xl-4", "po-offset-lg-4", "po-offset-md-3", "po-lg-4", "po-md-6", 3, "p-disabled", "p-label", "p-loading", "p-click"], [1, "po-table"], ["class", "po-table-column-selectable", 4, "ngIf"], ["class", "po-table-header-column po-table-header-master-detail", 4, "ngIf"], [3, "po-table-header-master-detail", "po-table-header-single-action", 4, "ngIf"], ["class", "po-table-header-column po-text-center", 4, "ngIf"], ["class", "po-table-header-ellipsis", 3, "width", "max-width", "min-width", "po-clickable", "po-table-header-subtitle", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], [3, "po-table-header-single-action", "po-table-header-actions", 4, "ngIf"], [3, "po-table-header-column-manager", "po-table-header-column-manager-border", "po-table-header-single-action", 4, "ngIf"], ["class", "po-table-group-row", 4, "ngIf"], [1, "po-table-column-selectable"], ["type", "checkbox", "class", "po-table-checkbox", 3, "po-table-checkbox-checked", "po-table-checkbox-indeterminate", 4, "ngIf"], ["class", "po-table-checkbox-label po-clickable", 3, "click", 4, "ngIf"], ["type", "checkbox", 1, "po-table-checkbox"], [1, "po-table-checkbox-label", "po-clickable", 3, "click"], [1, "po-table-header-column", "po-table-header-master-detail"], [1, "po-table-header-column", "po-text-center"], ["noColumnsHeader", ""], [4, "ngIf", "ngIfThen", "ngIfElse"], [1, "po-table-header-ellipsis", 3, "click"], ["headersTable", ""], [1, "po-table-header-flex"], [4, "ngTemplateOutlet", "ngTemplateOutletContext"], ["columnManager", ""], ["p-tooltip-position", "left", 1, "po-table-header-column-manager-button", "po-icon", "po-icon-settings", "po-clickable", 3, "p-tooltip", "click"], ["columnManagerTarget", ""], [1, "po-table-group-row"], [1, "po-table-row"], [1, "po-table-no-data", "po-text-center", 3, "colSpan"], ["class", "po-table-group-row", 4, "ngFor", "ngForOf", "ngForTrackBy"], ["class", "po-table-column po-table-column-selectable", 4, "ngIf"], ["class", "po-table-column-detail-toggle", 3, "click", 4, "ngIf"], [3, "width", "max-width", "min-width", "po-table-column", "po-table-column-right", "po-table-column-center", "po-table-column-icons", "ngClass", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], ["class", "po-table-column", 4, "ngIf"], [1, "po-table-column", "po-table-column-selectable"], [1, "po-table-column-detail-toggle", 3, "click"], [3, "ngTemplateOutlet", "ngTemplateOutletContext"], [3, "ngClass", "click"], [1, "po-table-column-cell", "notranslate", 3, "ngSwitch", "p-tooltip", "mouseenter", "mouseleave"], [4, "ngSwitchCase"], [3, "p-action", "p-disabled", "p-link", "p-row", "p-value", "click", 4, "ngSwitchCase"], [3, "p-column", "p-icons", "p-row", 4, "ngSwitchCase"], [4, "ngSwitchDefault"], [3, "p-action", "p-disabled", "p-link", "p-row", "p-value", "click"], [3, "p-column", "p-icons", "p-row"], [3, "p-subtitle"], [3, "p-value"], [1, "po-table-column"], [1, "po-table-row-template-container", 3, "colSpan"], [1, "po-table-column-detail", 3, "colSpan"], [3, "p-selectable", "p-detail", "p-items", "p-select-row"], ["class", "po-icon po-clickable", 3, "po-icon-arrow-up", "po-icon-arrow-down", 4, "ngIf"], [1, "po-icon", "po-clickable"], ["type", "radio", 1, "po-table-radio"], [1, "po-table-radio-label", "po-clickable", 3, "click"], [1, "po-table-header-ellipsis", 3, "p-tooltip", "mouseenter", "mouseleave"], ["columnHeader", ""], [3, "po-table-header-icon-unselected", "po-table-header-icon-descending", "po-table-header-icon-ascending", 4, "ngIf"], [1, "po-table-header-fixed-inner"], ["class", "po-table-column po-table-column-single-action", 4, "ngIf"], ["class", "po-table-column-actions", 4, "ngIf"], [1, "po-table-column", "po-table-column-single-action"], ["class", "po-table-single-action po-clickable", 3, "po-table-action-disabled", "click", 4, "ngIf"], [1, "po-table-single-action", "po-clickable", 3, "click"], ["class", "po-table-single-action-content", 3, "p-icon", 4, "ngIf"], [1, "po-table-single-action-content", 3, "p-icon"], [1, "po-table-column-actions"], [1, "po-icon", "po-icon-more", "po-clickable", 3, "click"], ["popupTarget", ""], [3, "p-columns", "p-max-columns", "p-target", "p-last-visible-columns-selected", "p-visible-columns-change", "p-change-visible-columns"]], template: function PoTableComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoTableComponent_po_container_0_Template, 2, 2, "po-container", 0);
        i0.ɵɵtemplate(1, PoTableComponent_ng_template_1_Template, 8, 10, "ng-template", null, 1, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(3, PoTableComponent_div_3_Template, 3, 5, "div", 2);
        i0.ɵɵtemplate(4, PoTableComponent_ng_template_4_Template, 13, 17, "ng-template", null, 3, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵelement(6, "po-popup", 4, 5);
        i0.ɵɵtemplate(8, PoTableComponent_ng_template_8_Template, 1, 1, "ng-template", null, 6, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(10, PoTableComponent_ng_template_10_Template, 2, 2, "ng-template", null, 7, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(12, PoTableComponent_ng_template_12_Template, 2, 2, "ng-template", null, 8, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(14, PoTableComponent_ng_template_14_Template, 5, 5, "ng-template", null, 9, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(16, PoTableComponent_ng_template_16_Template, 2, 3, "ng-template", null, 10, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(18, PoTableComponent_ng_template_18_Template, 1, 1, "ng-template", null, 11, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(20, PoTableComponent_ng_template_20_Template, 2, 2, "ng-template", null, 12, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(22, PoTableComponent_po_table_column_manager_22_Template, 1, 4, "po-table-column-manager", 13);
    } if (rf & 2) {
        const _r1 = i0.ɵɵreference(2);
        i0.ɵɵproperty("ngIf", ctx.container)("ngIfElse", _r1);
        i0.ɵɵadvance(3);
        i0.ɵɵproperty("ngIf", !ctx.infiniteScroll);
        i0.ɵɵadvance(3);
        i0.ɵɵproperty("p-actions", ctx.actions)("p-target", ctx.popupTarget);
        i0.ɵɵadvance(16);
        i0.ɵɵproperty("ngIf", !ctx.hideColumnsManager);
    } }, directives: [i3.NgIf, i6.PoContainerComponent, i3.NgTemplateOutlet, i7.PoLoadingComponent, i3.NgForOf, i8.PoTableSubtitleFooterComponent, i9.PoButtonComponent, i10.PoTooltipDirective, i3.NgClass, i3.NgSwitch, i3.NgSwitchCase, i11.PoTableColumnLinkComponent, i12.PoTableColumnIconComponent, i13.PoTableSubtitleCircleComponent, i14.PoTableColumnLabelComponent, i3.NgSwitchDefault, i15.PoTableDetailComponent, i16.PoPopupComponent, i17.PoIconComponent, i18.PoTableColumnManagerComponent], pipes: [i3.CurrencyPipe, i3.DatePipe, i19.PoTimePipe, i3.TitleCasePipe], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTableComponent, [{
        type: Component,
        args: [{ selector: 'po-table', providers: [PoDateService], template: "<po-container *ngIf=\"container; else tableContainerTemplate\" p-no-padding [p-no-shadow]=\"container === 'border'\">\n  <ng-container *ngTemplateOutlet=\"tableContainerTemplate\"></ng-container>\n</po-container>\n\n<ng-template #tableContainerTemplate>\n  <div [class.po-table-container-relative]=\"loading\">\n    <div *ngIf=\"loading\" class=\"po-table-overlay\">\n      <po-loading class=\"po-table-overlay-content\" [p-text]=\"literals.loadingData\"></po-loading>\n    </div>\n\n    <div class=\"po-table-main-container\">\n      <div\n        #tableWrapper\n        class=\"po-table-wrapper\"\n        [class.po-table-header-fixed-columns-pixels]=\"allColumnsWidthPixels\"\n        [style.opacity]=\"tableOpacity\"\n      >\n        <div *ngIf=\"height\" class=\"po-table-container\" [style.height.px]=\"heightTableContainer\">\n          <div class=\"po-table-header-fixed po-table-header\"></div>\n          <div #poTableTbody class=\"po-table-container-fixed-inner\">\n            <ng-container *ngTemplateOutlet=\"tableTemplate\"></ng-container>\n          </div>\n        </div>\n\n        <div *ngIf=\"!height\">\n          <ng-container *ngTemplateOutlet=\"tableTemplate\"></ng-container>\n        </div>\n      </div>\n    </div>\n  </div>\n\n  <div class=\"po-table-footer\" *ngIf=\"hasFooter\">\n    <ng-container *ngFor=\"let column of subtitleColumns; trackBy: trackBy\">\n      <po-table-subtitle-footer [p-literals]=\"literals\" [p-subtitles]=\"column.subtitles\"> </po-table-subtitle-footer>\n    </ng-container>\n  </div>\n</ng-template>\n\n<div\n  *ngIf=\"!infiniteScroll\"\n  class=\"po-row po-table-footer-show-more\"\n  [class.po-invisible]=\"showMore.observers.length === 0 && !hasService\"\n  #tableFooter\n>\n  <po-button\n    class=\"po-offset-xl-4 po-offset-lg-4 po-offset-md-3 po-lg-4 po-md-6\"\n    [p-disabled]=\"showMoreDisabled\"\n    [p-label]=\"literals.loadMoreData\"\n    [p-loading]=\"loadingShowMore\"\n    (p-click)=\"onShowMore()\"\n  >\n  </po-button>\n</div>\n\n<ng-template #tableTemplate>\n  <table class=\"po-table\" [class.po-table-striped]=\"striped\" [class.po-table-layout-fixed]=\"hideTextOverflow\">\n    <thead>\n      <tr [class.po-table-header]=\"!height\">\n        <th *ngIf=\"hasSelectableColumn\" class=\"po-table-column-selectable\">\n          <div [class.po-table-header-fixed-inner]=\"height\">\n            <input\n              *ngIf=\"!hideSelectAll\"\n              type=\"checkbox\"\n              class=\"po-table-checkbox\"\n              [class.po-table-checkbox-checked]=\"selectAll\"\n              [class.po-table-checkbox-indeterminate]=\"selectAll === null\"\n            />\n            <label *ngIf=\"!hideSelectAll\" class=\"po-table-checkbox-label po-clickable\" (click)=\"selectAllRows()\">\n            </label>\n          </div>\n        </th>\n\n        <th\n          *ngIf=\"(hasMasterDetailColumn || hasRowTemplate) && !hasRowTemplateWithArrowDirectionRight\"\n          class=\"po-table-header-column po-table-header-master-detail\"\n        ></th>\n\n        <!-- Coluna criada para caso as a\u00E7\u00F5es fiquem no lado esquerdo -->\n        <th\n          *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\n          [class.po-table-header-master-detail]=\"!isSingleAction\"\n          [class.po-table-header-single-action]=\"isSingleAction\"\n        ></th>\n\n        <th *ngIf=\"!hasMainColumns\" #noColumnsHeader class=\"po-table-header-column po-text-center\">\n          <ng-container *ngIf=\"height; then noColumnsWithHeight; else noColumnsWithoutHeight\"> </ng-container>\n        </th>\n\n        <th\n          *ngFor=\"let column of mainColumns; let i = index; trackBy: trackBy\"\n          #headersTable\n          class=\"po-table-header-ellipsis\"\n          [style.width]=\"column.width\"\n          [style.max-width]=\"column.width\"\n          [style.min-width]=\"column.width\"\n          [class.po-clickable]=\"(sort && column.sortable !== false) || hasService\"\n          [class.po-table-header-subtitle]=\"column.type === 'subtitle'\"\n          (click)=\"sortColumn(column)\"\n        >\n          <div\n            class=\"po-table-header-flex\"\n            [class.po-table-header-fixed-inner]=\"height\"\n            [class.po-table-header-flex-right]=\"column.type === 'currency' || column.type === 'number'\"\n            [class.po-table-header-flex-center]=\"column.type === 'subtitle'\"\n          >\n            <ng-container *ngTemplateOutlet=\"contentHeaderTemplate; context: { $implicit: column }\"> </ng-container>\n          </div>\n        </th>\n\n        <th\n          *ngIf=\"hasRowTemplateWithArrowDirectionRight && (hasVisibleActions || hideColumnsManager)\"\n          class=\"po-table-header-column po-table-header-master-detail\"\n        ></th>\n\n        <th\n          *ngIf=\"hasVisibleActions && hideColumnsManager && actionRight\"\n          [class.po-table-header-single-action]=\"isSingleAction\"\n          [class.po-table-header-actions]=\"!isSingleAction\"\n        ></th>\n\n        <th\n          #columnManager\n          *ngIf=\"hasValidColumns && !hideColumnsManager\"\n          [class.po-table-header-column-manager]=\"!isSingleAction || !actionRight\"\n          [class.po-table-header-column-manager-border]=\"!height && container\"\n          [class.po-table-header-single-action]=\"isSingleAction && actionRight\"\n        >\n          <div\n            [class.po-table-header-column-manager-border]=\"height && container\"\n            [class.po-table-header-column-manager-fixed-inner]=\"height\"\n            [style.width.px]=\"height && visibleActions.length ? columnManager.offsetWidth : undefined\"\n          >\n            <button\n              #columnManagerTarget\n              class=\"po-table-header-column-manager-button po-icon po-icon-settings po-clickable\"\n              p-tooltip-position=\"left\"\n              [p-tooltip]=\"literals.columnsManager\"\n              (click)=\"onOpenColumnManager()\"\n            ></button>\n          </div>\n        </th>\n      </tr>\n    </thead>\n\n    <tbody class=\"po-table-group-row\" *ngIf=\"!hasItems || !hasMainColumns\">\n      <tr class=\"po-table-row\">\n        <td [colSpan]=\"columnCount\" class=\"po-table-no-data po-text-center\">\n          <span> {{ literals.noData }} </span>\n        </td>\n      </tr>\n    </tbody>\n\n    <ng-container *ngIf=\"hasMainColumns\">\n      <tbody class=\"po-table-group-row\" *ngFor=\"let row of items; let rowIndex = index; trackBy: trackBy\">\n        <tr class=\"po-table-row\" [class.po-table-row-active]=\"row.$selected || (row.$selected === null && selectable)\">\n          <td *ngIf=\"selectable\" class=\"po-table-column po-table-column-selectable\">\n            <ng-container *ngTemplateOutlet=\"singleSelect ? inputRadio : inputCheckbox; context: { $implicit: row }\">\n            </ng-container>\n          </td>\n\n          <!-- Valida se a origem do detail \u00E9 pelo input do po-table -->\n          <td\n            *ngIf=\"columnMasterDetail && !hideDetail && !hasRowTemplate\"\n            class=\"po-table-column-detail-toggle\"\n            (click)=\"toggleDetail(row)\"\n          >\n            <ng-template\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\n            >\n            </ng-template>\n          </td>\n\n          <!-- Coluna com as a\u00E7\u00F5es na esquerda (padr\u00E3o)-->\n          <ng-template\n            *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\n            [ngTemplateOutlet]=\"ActionsColumnTemplate\"\n            [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\n          >\n          </ng-template>\n\n          <!-- Valida se a origem do detail \u00E9 pela diretiva -->\n          <td\n            *ngIf=\"hasRowTemplate && !hasRowTemplateWithArrowDirectionRight\"\n            class=\"po-table-column-detail-toggle\"\n            (click)=\"toggleDetail(row)\"\n          >\n            <ng-template\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\n            >\n            </ng-template>\n          </td>\n\n          <td\n            *ngFor=\"let column of mainColumns; let columnIndex = index; trackBy: trackBy\"\n            [style.width]=\"column.width\"\n            [style.max-width]=\"column.width\"\n            [style.min-width]=\"column.width\"\n            [class.po-table-column]=\"column.type !== 'icon'\"\n            [class.po-table-column-right]=\"column.type === 'currency' || column.type === 'number'\"\n            [class.po-table-column-center]=\"column.type === 'subtitle'\"\n            [class.po-table-column-icons]=\"column.type === 'icon'\"\n            [ngClass]=\"getClassColor(row, column)\"\n            (click)=\"selectable ? selectRow(row) : 'javascript:;'\"\n          >\n            <div\n              class=\"po-table-column-cell notranslate\"\n              [class.po-table-body-ellipsis]=\"hideTextOverflow\"\n              [ngSwitch]=\"column.type\"\n              [p-tooltip]=\"tooltipText\"\n              (mouseenter)=\"tooltipMouseEnter($event, column, row)\"\n              (mouseleave)=\"tooltipMouseLeave()\"\n            >\n              <span *ngSwitchCase=\"'columnTemplate'\">\n                <ng-container *ngTemplateOutlet=\"getTemplate(column); context: { $implicit: getCellData(row, column) }\">\n                </ng-container>\n              </span>\n\n              <span *ngSwitchCase=\"'cellTemplate'\">\n                <ng-container *ngTemplateOutlet=\"tableCellTemplate?.templateRef; context: { row: row, column: column }\">\n                </ng-container>\n              </span>\n\n              <span *ngSwitchCase=\"'boolean'\">\n                {{ getBooleanLabel(getCellData(row, column), column) }}\n              </span>\n\n              <span *ngSwitchCase=\"'currency'\">\n                {{ getCellData(row, column) | currency: column.format:'symbol':'1.2-2' }}\n              </span>\n\n              <span *ngSwitchCase=\"'date'\">\n                {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy' }}\n              </span>\n\n              <span *ngSwitchCase=\"'time'\">\n                {{ getCellData(row, column) | po_time: column.format || 'HH:mm:ss.ffffff' }}\n              </span>\n\n              <span *ngSwitchCase=\"'dateTime'\">\n                {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy HH:mm:ss' }}\n              </span>\n\n              <span *ngSwitchCase=\"'number'\">\n                {{ formatNumber(getCellData(row, column), column.format) }}\n              </span>\n\n              <po-table-column-link\n                *ngSwitchCase=\"'link'\"\n                [p-action]=\"column.action\"\n                [p-disabled]=\"checkDisabled(row, column)\"\n                [p-link]=\"row[column.link]\"\n                [p-row]=\"row\"\n                [p-value]=\"getCellData(row, column)\"\n                (click)=\"onClickLink($event, row, column)\"\n              >\n              </po-table-column-link>\n\n              <po-table-column-icon\n                *ngSwitchCase=\"'icon'\"\n                [p-column]=\"column\"\n                [p-icons]=\"getColumnIcons(row, column)\"\n                [p-row]=\"row\"\n              >\n              </po-table-column-icon>\n\n              <span *ngSwitchCase=\"'subtitle'\">\n                <po-table-subtitle-circle [p-subtitle]=\"getSubtitleColumn(row, column)\"></po-table-subtitle-circle>\n              </span>\n              <span *ngSwitchCase=\"'label'\">\n                <po-table-column-label [p-value]=\"getColumnLabel(row, column)\"> </po-table-column-label>\n              </span>\n              <span *ngSwitchDefault>{{ getCellData(row, column) }}</span>\n            </div>\n          </td>\n\n          <td\n            *ngIf=\"hasRowTemplateWithArrowDirectionRight\"\n            class=\"po-table-column-detail-toggle\"\n            (click)=\"toggleDetail(row)\"\n          >\n            <ng-template\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\n            >\n            </ng-template>\n          </td>\n\n          <!-- Coluna de a\u00E7oes na direita -->\n          <ng-template\n            *ngIf=\"actionRight\"\n            [ngTemplateOutlet]=\"ActionsColumnTemplate\"\n            [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\n          >\n          </ng-template>\n\n          <!-- Coluna para n\u00E3o ficar em branco nas linhas de gerenciamento -->\n          <ng-container *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction) && !hideColumnsManager\">\n            <td class=\"po-table-column\"></td>\n          </ng-container>\n\n          <!-- Column Manager -->\n          <td\n            *ngIf=\"!hasVisibleActions && !hideColumnsManager && !hasRowTemplateWithArrowDirectionRight\"\n            class=\"po-table-column\"\n          ></td>\n        </tr>\n\n        <tr *ngIf=\"hasMainColumns && hasRowTemplate && row.$showDetail && isShowRowTemplate(row, rowIndex)\">\n          <td class=\"po-table-row-template-container\" [colSpan]=\"columnCountForMasterDetail\">\n            <ng-template\n              [ngTemplateOutlet]=\"tableRowTemplate.templateRef\"\n              [ngTemplateOutletContext]=\"{ $implicit: row, rowIndex: rowIndex }\"\n            >\n            </ng-template>\n          </td>\n        </tr>\n\n        <tr *ngIf=\"hasMainColumns && isShowMasterDetail(row)\">\n          <td class=\"po-table-column-detail\" [colSpan]=\"columnCountForMasterDetail\">\n            <po-table-detail\n              [p-selectable]=\"selectable && !detailHideSelect\"\n              [p-detail]=\"columnMasterDetail.detail\"\n              [p-items]=\"row[nameColumnDetail]\"\n              (p-select-row)=\"selectDetailRow($event)\"\n            >\n            </po-table-detail>\n          </td>\n        </tr>\n      </tbody>\n    </ng-container>\n  </table>\n</ng-template>\n\n<po-popup #popup [p-actions]=\"actions\" [p-target]=\"popupTarget\"> </po-popup>\n\n<ng-template #poTableColumnDetail let-row=\"row\" let-rowIndex=\"rowIndex\">\n  <span\n    *ngIf=\"(containsMasterDetail(row) && !hasRowTemplate) || (isShowRowTemplate(row, rowIndex) && hasRowTemplate)\"\n    class=\"po-icon po-clickable\"\n    [class.po-icon-arrow-up]=\"row.$showDetail\"\n    [class.po-icon-arrow-down]=\"!row.$showDetail\"\n  >\n  </span>\n</ng-template>\n\n<ng-template #inputRadio let-row>\n  <input type=\"radio\" class=\"po-table-radio\" [class.po-table-radio-checked]=\"row.$selected\" />\n  <label class=\"po-table-radio-label po-clickable\" (click)=\"selectable ? selectRow(row) : 'javascript:;'\"></label>\n</ng-template>\n\n<ng-template #inputCheckbox let-row>\n  <input type=\"checkbox\" class=\"po-table-checkbox\" [class.po-table-checkbox-checked]=\"row.$selected\" />\n  <label class=\"po-table-checkbox-label po-clickable\" (click)=\"selectable ? selectRow(row) : 'javascript:;'\"></label>\n</ng-template>\n\n<ng-template #contentHeaderTemplate let-column>\n  <span\n    #columnHeader\n    class=\"po-table-header-ellipsis\"\n    [p-tooltip]=\"tooltipText\"\n    (mouseenter)=\"tooltipMouseEnter($event)\"\n    (mouseleave)=\"tooltipMouseLeave()\"\n  >\n    {{ column.label || (column.property | titlecase) }}\n  </span>\n  <span\n    *ngIf=\"sort && column.sortable !== false\"\n    [class.po-table-header-icon-unselected]=\"sortedColumn?.property !== column\"\n    [class.po-table-header-icon-descending]=\"sortedColumn?.property === column && sortedColumn.ascending\"\n    [class.po-table-header-icon-ascending]=\"sortedColumn?.property === column && !sortedColumn.ascending\"\n  >\n  </span>\n</ng-template>\n\n<ng-template #noColumnsWithHeight>\n  <div class=\"po-table-header-fixed-inner\" [style.width.px]=\"noColumnsHeader?.nativeElement.offsetWidth\">\n    {{ hasValidColumns ? literals.noVisibleColumn : literals.noColumns }}\n  </div>\n</ng-template>\n\n<ng-template #noColumnsWithoutHeight>\n  {{ hasValidColumns ? literals.noVisibleColumn : literals.noColumns }}\n</ng-template>\n\n<!-- Template de a\u00E7\u00F5es -->\n<ng-template #ActionsColumnTemplate let-row=\"row\" let-rowIndex=\"rowIndex\">\n  <td *ngIf=\"isSingleAction\" class=\"po-table-column po-table-column-single-action\">\n    <div\n      *ngIf=\"firstAction.visible !== false\"\n      class=\"po-table-single-action po-clickable\"\n      [class.po-table-action-disabled]=\"firstAction.disabled ? validateTableAction(row, firstAction) : false\"\n      (click)=\"executeTableAction(row, firstAction)\"\n    >\n      <po-icon *ngIf=\"firstAction.icon\" class=\"po-table-single-action-content\" [p-icon]=\"firstAction.icon\"></po-icon>\n      {{ firstAction.label }}\n    </div>\n  </td>\n\n  <td *ngIf=\"visibleActions.length > 1\" class=\"po-table-column-actions\">\n    <span #popupTarget class=\"po-icon po-icon-more po-clickable\" (click)=\"togglePopup(row, popupTarget)\"></span>\n  </td>\n</ng-template>\n\n<po-table-column-manager\n  *ngIf=\"!hideColumnsManager\"\n  [p-columns]=\"columns\"\n  [p-max-columns]=\"maxColumns\"\n  [p-target]=\"columnManagerTarget\"\n  [p-last-visible-columns-selected]=\"lastVisibleColumnsSelected\"\n  (p-visible-columns-change)=\"onVisibleColumnsChange($event)\"\n  (p-change-visible-columns)=\"onChangeVisibleColumns($event)\"\n>\n</po-table-column-manager>\n" }]
    }], function () { return [{ type: i1.PoDateService }, { type: i0.IterableDiffers }, { type: i0.Renderer2 }, { type: i2.PoLanguageService }, { type: i0.ChangeDetectorRef }, { type: i3.DecimalPipe }, { type: i4.Router }, { type: i5.PoTableService }]; }, { tableRowTemplate: [{
            type: ContentChild,
            args: [PoTableRowTemplateDirective, { static: true }]
        }], tableCellTemplate: [{
            type: ContentChild,
            args: [PoTableCellTemplateDirective]
        }], tableColumnTemplates: [{
            type: ContentChildren,
            args: [PoTableColumnTemplateDirective]
        }], noColumnsHeader: [{
            type: ViewChild,
            args: ['noColumnsHeader', { read: ElementRef }]
        }], poPopupComponent: [{
            type: ViewChild,
            args: ['popup']
        }], tableFooterElement: [{
            type: ViewChild,
            args: ['tableFooter', { read: ElementRef, static: false }]
        }], tableWrapperElement: [{
            type: ViewChild,
            args: ['tableWrapper', { read: ElementRef, static: false }]
        }], poTableTbody: [{
            type: ViewChild,
            args: ['poTableTbody', { read: ElementRef, static: false }]
        }], actionsIconElement: [{
            type: ViewChildren,
            args: ['actionsIconElement', { read: ElementRef }]
        }], actionsElement: [{
            type: ViewChildren,
            args: ['actionsElement', { read: ElementRef }]
        }], headersTable: [{
            type: ViewChildren,
            args: ['headersTable']
        }], columnManagerTarget: [{
            type: ViewChild,
            args: ['columnManagerTarget']
        }] }); })();
