// @po-ui/ng-templates/esm2020/lib/components/po-page-change-password/po-page-change-password.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoModule } from '@po-ui/ng-components';
import { PoPageBackgroundModule } from '../po-page-background/index';
import { PoPageChangePasswordComponent } from './po-page-change-password.component';
import { PoPageChangePasswordService } from './po-page-change-password.service';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-change-password.
 */
export class PoPageChangePasswordModule {
}
PoPageChangePasswordModule.ɵfac = function PoPageChangePasswordModule_Factory(t) { return new (t || PoPageChangePasswordModule)(); };
PoPageChangePasswordModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageChangePasswordModule });
PoPageChangePasswordModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [PoPageChangePasswordService], imports: [[CommonModule, FormsModule, RouterModule, PoModule, PoPageBackgroundModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageChangePasswordModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, RouterModule, PoModule, PoPageBackgroundModule],
                declarations: [PoPageChangePasswordComponent],
                providers: [PoPageChangePasswordService],
                exports: [PoPageChangePasswordComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageChangePasswordModule, { declarations: [PoPageChangePasswordComponent], imports: [CommonModule, FormsModule, RouterModule, PoModule, PoPageBackgroundModule], exports: [PoPageChangePasswordComponent] }); })();
