// @po-ui/ng-components/esm2020/lib/components/po-field/po-select/po-select-option-template/po-select-option-template.directive.mjs
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @usedBy PoSelectComponent
 *
 * @deprecated 14.x.x
 *
 * @description
 *
 * > **Deprecated 14.x.x**. Utilize o `PoComboOptionTemplate` em conjunto com o componente `po-combo`.
 *
 * Esta diretiva permite personalizar o conteúdo dos itens exibidos na lista suspensa do componente.
 *
 * Quando utilizada em _mobile_ será exibido o componente nativo.
 *
 * Para personalizar o conteúdo de cada item da lista deve-se utilizar a diretiva `p-select-option-template` com `ng-template`
 * dentro da tag `po-select`.
 *
 * Para obter a referência do item atual, use `let-option`, com isso você terá acesso aos valores e poderá personalizar sua exibição.
 *
 * Esta diretiva compõe-se de dois meios para uso, de forma explícita tal como em *syntax sugar*. Veja a seguir ambos, respectivamente:
 *
 * ```
 * ...
 * <po-select
 *   name="select"
 *   [(ngModel)]="select"
 *   [p-options]="options">
 *     <ng-template p-select-option-template let-option>
 *       <option-template [option]="option"></option-template>
 *     </ng-template>
 * </po-select>
 * ...
 * ```
 *
 * ```
 * ...
 * <po-select
 *   name="select"
 *   [(ngModel)]="select"
 *   [p-options]="options">
 *     <div *p-select-option-template="let option">
 *       <option-template [option]="option"></option-template>
 *     </div>
 * </po-select>
 * ...
 *
 * ```
 */
export class PoSelectOptionTemplateDirective {
    // Necessário manter templateRef para o funcionamento do row template.
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
PoSelectOptionTemplateDirective.ɵfac = function PoSelectOptionTemplateDirective_Factory(t) { return new (t || PoSelectOptionTemplateDirective)(i0.ɵɵdirectiveInject(i0.TemplateRef)); };
PoSelectOptionTemplateDirective.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: PoSelectOptionTemplateDirective, selectors: [["", "p-select-option-template", ""]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoSelectOptionTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[p-select-option-template]'
            }]
    }], function () { return [{ type: i0.TemplateRef }]; }, null); })();
