// @po-ui/ng-components/esm2020/lib/components/components.module.mjs
import { NgModule } from '@angular/core';
import { PoAccordionModule } from './po-accordion/po-accordion.module';
import { PoAvatarModule } from './po-avatar/po-avatar.module';
import { PoBreadcrumbModule } from './po-breadcrumb/po-breadcrumb.module';
import { PoButtonModule } from './po-button/po-button.module';
import { PoButtonGroupModule } from './po-button-group/po-button-group.module';
import { PoCalendarModule } from './po-calendar/po-calendar.module';
import { PoChartModule } from './po-chart/po-chart.module';
import { PoContainerModule } from './po-container/po-container.module';
import { PoDisclaimerGroupModule } from './po-disclaimer-group/po-disclaimer-group.module';
import { PoDisclaimerModule } from './po-disclaimer/po-disclaimer.module';
import { PoDividerModule } from './po-divider/po-divider.module';
import { PoDropdownModule } from './po-dropdown/po-dropdown.module';
import { PoDynamicModule } from './po-dynamic/po-dynamic.module';
import { PoFieldModule } from './po-field/po-field.module';
import { PoGaugeModule } from './po-gauge/po-gauge.module';
import { PoGridModule } from './po-grid/po-grid.module';
import { PoInfoModule } from './po-info/po-info.module';
import { PoListViewModule } from './po-list-view/po-list-view.module';
import { PoLoadingModule } from './po-loading/po-loading.module';
import { PoMenuModule } from './po-menu/po-menu.module';
import { PoMenuPanelModule } from './po-menu-panel/po-menu-panel.module';
import { PoModalModule } from './po-modal/po-modal.module';
import { PoNavbarModule } from './po-navbar/po-navbar.module';
import { PoPageModule } from './po-page/po-page.module';
import { PoPopoverModule } from './po-popover/po-popover.module';
import { PoPopupModule } from './po-popup/po-popup.module';
import { PoProgressModule } from './po-progress/po-progress.module';
import { PoSlideModule } from './po-slide/po-slide.module';
import { PoStepperModule } from './po-stepper/po-stepper.module';
import { PoTableModule } from './po-table/po-table.module';
import { PoTabsModule } from './po-tabs/po-tabs.module';
import { PoTagModule } from './po-tag/po-tag.module';
import { PoToolbarModule } from './po-toolbar/po-toolbar.module';
import { PoTreeViewModule } from './po-tree-view/po-tree-view.module';
import { PoWidgetModule } from './po-widget/po-widget.module';
import { PoIconModule } from './po-icon/po-icon.module';
import * as i0 from "@angular/core";
export class PoComponentsModule {
}
PoComponentsModule.ɵfac = function PoComponentsModule_Factory(t) { return new (t || PoComponentsModule)(); };
PoComponentsModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoComponentsModule });
PoComponentsModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [], imports: [[
            PoAccordionModule,
            PoAvatarModule,
            PoBreadcrumbModule,
            PoButtonModule,
            PoButtonGroupModule,
            PoCalendarModule,
            PoChartModule,
            PoContainerModule,
            PoDisclaimerGroupModule,
            PoDisclaimerModule,
            PoDividerModule,
            PoDropdownModule,
            PoDynamicModule,
            PoFieldModule,
            PoGaugeModule,
            PoGridModule,
            PoIconModule,
            PoInfoModule,
            PoListViewModule,
            PoLoadingModule,
            PoMenuModule,
            PoMenuPanelModule,
            PoModalModule,
            PoNavbarModule,
            PoPageModule,
            PoPopoverModule,
            PoPopupModule,
            PoProgressModule,
            PoSlideModule,
            PoStepperModule,
            PoTableModule,
            PoTabsModule,
            PoTagModule,
            PoToolbarModule,
            PoTreeViewModule,
            PoWidgetModule
        ], PoAccordionModule,
        PoAvatarModule,
        PoBreadcrumbModule,
        PoButtonModule,
        PoButtonGroupModule,
        PoCalendarModule,
        PoChartModule,
        PoContainerModule,
        PoDisclaimerGroupModule,
        PoDisclaimerModule,
        PoDividerModule,
        PoDropdownModule,
        PoDynamicModule,
        PoFieldModule,
        PoGaugeModule,
        PoGridModule,
        PoIconModule,
        PoInfoModule,
        PoListViewModule,
        PoLoadingModule,
        PoMenuModule,
        PoMenuPanelModule,
        PoModalModule,
        PoNavbarModule,
        PoPageModule,
        PoPopoverModule,
        PoPopupModule,
        PoProgressModule,
        PoSlideModule,
        PoStepperModule,
        PoTableModule,
        PoTabsModule,
        PoTagModule,
        PoToolbarModule,
        PoTreeViewModule,
        PoWidgetModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoComponentsModule, [{
        type: NgModule,
        args: [{
                imports: [
                    PoAccordionModule,
                    PoAvatarModule,
                    PoBreadcrumbModule,
                    PoButtonModule,
                    PoButtonGroupModule,
                    PoCalendarModule,
                    PoChartModule,
                    PoContainerModule,
                    PoDisclaimerGroupModule,
                    PoDisclaimerModule,
                    PoDividerModule,
                    PoDropdownModule,
                    PoDynamicModule,
                    PoFieldModule,
                    PoGaugeModule,
                    PoGridModule,
                    PoIconModule,
                    PoInfoModule,
                    PoListViewModule,
                    PoLoadingModule,
                    PoMenuModule,
                    PoMenuPanelModule,
                    PoModalModule,
                    PoNavbarModule,
                    PoPageModule,
                    PoPopoverModule,
                    PoPopupModule,
                    PoProgressModule,
                    PoSlideModule,
                    PoStepperModule,
                    PoTableModule,
                    PoTabsModule,
                    PoTagModule,
                    PoToolbarModule,
                    PoTreeViewModule,
                    PoWidgetModule
                ],
                exports: [
                    PoAccordionModule,
                    PoAvatarModule,
                    PoBreadcrumbModule,
                    PoButtonModule,
                    PoButtonGroupModule,
                    PoCalendarModule,
                    PoChartModule,
                    PoContainerModule,
                    PoDisclaimerGroupModule,
                    PoDisclaimerModule,
                    PoDividerModule,
                    PoDropdownModule,
                    PoDynamicModule,
                    PoFieldModule,
                    PoGaugeModule,
                    PoGridModule,
                    PoIconModule,
                    PoInfoModule,
                    PoListViewModule,
                    PoLoadingModule,
                    PoMenuModule,
                    PoMenuPanelModule,
                    PoModalModule,
                    PoNavbarModule,
                    PoPageModule,
                    PoPopoverModule,
                    PoPopupModule,
                    PoProgressModule,
                    PoSlideModule,
                    PoStepperModule,
                    PoTableModule,
                    PoTabsModule,
                    PoTagModule,
                    PoToolbarModule,
                    PoTreeViewModule,
                    PoWidgetModule
                ],
                providers: [],
                bootstrap: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoComponentsModule, { imports: [PoAccordionModule,
        PoAvatarModule,
        PoBreadcrumbModule,
        PoButtonModule,
        PoButtonGroupModule,
        PoCalendarModule,
        PoChartModule,
        PoContainerModule,
        PoDisclaimerGroupModule,
        PoDisclaimerModule,
        PoDividerModule,
        PoDropdownModule,
        PoDynamicModule,
        PoFieldModule,
        PoGaugeModule,
        PoGridModule,
        PoIconModule,
        PoInfoModule,
        PoListViewModule,
        PoLoadingModule,
        PoMenuModule,
        PoMenuPanelModule,
        PoModalModule,
        PoNavbarModule,
        PoPageModule,
        PoPopoverModule,
        PoPopupModule,
        PoProgressModule,
        PoSlideModule,
        PoStepperModule,
        PoTableModule,
        PoTabsModule,
        PoTagModule,
        PoToolbarModule,
        PoTreeViewModule,
        PoWidgetModule], exports: [PoAccordionModule,
        PoAvatarModule,
        PoBreadcrumbModule,
        PoButtonModule,
        PoButtonGroupModule,
        PoCalendarModule,
        PoChartModule,
        PoContainerModule,
        PoDisclaimerGroupModule,
        PoDisclaimerModule,
        PoDividerModule,
        PoDropdownModule,
        PoDynamicModule,
        PoFieldModule,
        PoGaugeModule,
        PoGridModule,
        PoIconModule,
        PoInfoModule,
        PoListViewModule,
        PoLoadingModule,
        PoMenuModule,
        PoMenuPanelModule,
        PoModalModule,
        PoNavbarModule,
        PoPageModule,
        PoPopoverModule,
        PoPopupModule,
        PoProgressModule,
        PoSlideModule,
        PoStepperModule,
        PoTableModule,
        PoTabsModule,
        PoTagModule,
        PoToolbarModule,
        PoTreeViewModule,
        PoWidgetModule] }); })();
