// @po-ui/ng-components/esm2020/lib/components/po-tag/po-tag.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoIconModule } from '../po-icon/po-icon.module';
import { PoTagComponent } from './po-tag.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-tag`.
 */
export class PoTagModule {
}
PoTagModule.ɵfac = function PoTagModule_Factory(t) { return new (t || PoTagModule)(); };
PoTagModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoTagModule });
PoTagModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [], imports: [[CommonModule, PoIconModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTagModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoIconModule],
                declarations: [PoTagComponent],
                exports: [PoTagComponent],
                providers: [],
                schemas: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoTagModule, { declarations: [PoTagComponent], imports: [CommonModule, PoIconModule], exports: [PoTagComponent] }); })();
