// @po-ui/ng-components/esm2020/lib/directives/po-tooltip/po-tooltip.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoTooltipDirective } from './po-tooltip.directive';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo da diretiva Po-Tooltip.
 */
export class PoTooltipModule {
}
PoTooltipModule.ɵfac = function PoTooltipModule_Factory(t) { return new (t || PoTooltipModule)(); };
PoTooltipModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoTooltipModule });
PoTooltipModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTooltipModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [PoTooltipDirective],
                exports: [PoTooltipDirective]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoTooltipModule, { declarations: [PoTooltipDirective], imports: [CommonModule], exports: [PoTooltipDirective] }); })();
