// @po-ui/ng-templates/esm2020/lib/components/po-page-dynamic-table/po-page-dynamic-table.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoTableModule } from '@po-ui/ng-components';
import { PoPageDynamicSearchModule } from '../po-page-dynamic-search/po-page-dynamic-search.module';
import { PoPageDynamicTableComponent } from './po-page-dynamic-table.component';
import { PoPageCustomizationModule } from '../../services/po-page-customization/po-page-customization.module';
import { PoPageDynamicModule } from '../../services/po-page-dynamic/po-page-dynamic.module';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-dynamic-table.
 */
export class PoPageDynamicTableModule {
}
PoPageDynamicTableModule.ɵfac = function PoPageDynamicTableModule_Factory(t) { return new (t || PoPageDynamicTableModule)(); };
PoPageDynamicTableModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageDynamicTableModule });
PoPageDynamicTableModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            FormsModule,
            RouterModule,
            PoTableModule,
            PoPageDynamicSearchModule,
            PoPageCustomizationModule,
            PoPageDynamicModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageDynamicTableModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    RouterModule,
                    PoTableModule,
                    PoPageDynamicSearchModule,
                    PoPageCustomizationModule,
                    PoPageDynamicModule
                ],
                declarations: [PoPageDynamicTableComponent],
                exports: [PoPageDynamicTableComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageDynamicTableModule, { declarations: [PoPageDynamicTableComponent], imports: [CommonModule,
        FormsModule,
        RouterModule,
        PoTableModule,
        PoPageDynamicSearchModule,
        PoPageCustomizationModule,
        PoPageDynamicModule], exports: [PoPageDynamicTableComponent] }); })();
