// @po-ui/ng-components/esm2020/lib/components/po-info/po-info.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoInfoComponent } from './po-info.component';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-info.
 */
export class PoInfoModule {
}
PoInfoModule.ɵfac = function PoInfoModule_Factory(t) { return new (t || PoInfoModule)(); };
PoInfoModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoInfoModule });
PoInfoModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [], imports: [[CommonModule, RouterModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoInfoModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule],
                declarations: [PoInfoComponent],
                exports: [PoInfoComponent],
                providers: [],
                schemas: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoInfoModule, { declarations: [PoInfoComponent], imports: [CommonModule, RouterModule], exports: [PoInfoComponent] }); })();
