// @po-ui/ng-templates/esm2020/lib/services/po-page-dynamic/po-page-dynamic.module.mjs
import { HttpClientModule } from '@angular/common/http';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoPageDynamicService } from './po-page-dynamic.service';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do serviço do po-page-dynamic-service.
 */
export class PoPageDynamicModule {
}
PoPageDynamicModule.ɵfac = function PoPageDynamicModule_Factory(t) { return new (t || PoPageDynamicModule)(); };
PoPageDynamicModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageDynamicModule });
PoPageDynamicModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [PoPageDynamicService], imports: [[CommonModule, HttpClientModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageDynamicModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, HttpClientModule],
                providers: [PoPageDynamicService]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageDynamicModule, { imports: [CommonModule, HttpClientModule] }); })();
