// @po-ui/ng-components/esm2020/lib/components/po-navbar/po-navbar-items/po-navbar-items.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoNavbarItemComponent } from './po-navbar-item/po-navbar-item.component';
import { PoNavbarItemsComponent } from './po-navbar-items.component';
import * as i0 from "@angular/core";
export class PoNavbarItemsModule {
}
PoNavbarItemsModule.ɵfac = function PoNavbarItemsModule_Factory(t) { return new (t || PoNavbarItemsModule)(); };
PoNavbarItemsModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoNavbarItemsModule });
PoNavbarItemsModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule, RouterModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNavbarItemsModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule],
                declarations: [PoNavbarItemComponent, PoNavbarItemsComponent],
                exports: [PoNavbarItemsComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoNavbarItemsModule, { declarations: [PoNavbarItemComponent, PoNavbarItemsComponent], imports: [CommonModule, RouterModule], exports: [PoNavbarItemsComponent] }); })();
