// @po-ui/ng-components/esm2020/lib/components/po-field/po-clean/po-clean.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoCleanComponent } from './po-clean.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-clean`.
 */
export class PoCleanModule {
}
PoCleanModule.ɵfac = function PoCleanModule_Factory(t) { return new (t || PoCleanModule)(); };
PoCleanModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoCleanModule });
PoCleanModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoCleanModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [PoCleanComponent],
                declarations: [PoCleanComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoCleanModule, { declarations: [PoCleanComponent], imports: [CommonModule], exports: [PoCleanComponent] }); })();
