// @po-ui/ng-templates/esm2020/lib/components/po-page-dynamic-search/po-page-dynamic-search.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoDynamicModule, PoLanguageModule, PoModalModule, PoPageModule } from '@po-ui/ng-components';
import { PoPageCustomizationModule } from '../../services/po-page-customization/po-page-customization.module';
import { PoAdvancedFilterComponent } from './po-advanced-filter/po-advanced-filter.component';
import { PoPageDynamicSearchComponent } from './po-page-dynamic-search.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-dynamic-search.
 */
export class PoPageDynamicSearchModule {
}
PoPageDynamicSearchModule.ɵfac = function PoPageDynamicSearchModule_Factory(t) { return new (t || PoPageDynamicSearchModule)(); };
PoPageDynamicSearchModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageDynamicSearchModule });
PoPageDynamicSearchModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            FormsModule,
            RouterModule,
            PoDynamicModule,
            PoLanguageModule,
            PoModalModule,
            PoPageModule,
            PoPageCustomizationModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageDynamicSearchModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    RouterModule,
                    PoDynamicModule,
                    PoLanguageModule,
                    PoModalModule,
                    PoPageModule,
                    PoPageCustomizationModule
                ],
                declarations: [PoAdvancedFilterComponent, PoPageDynamicSearchComponent],
                exports: [PoPageDynamicSearchComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageDynamicSearchModule, { declarations: [PoAdvancedFilterComponent, PoPageDynamicSearchComponent], imports: [CommonModule,
        FormsModule,
        RouterModule,
        PoDynamicModule,
        PoLanguageModule,
        PoModalModule,
        PoPageModule,
        PoPageCustomizationModule], exports: [PoPageDynamicSearchComponent] }); })();
