// @po-ui/ng-components/esm2020/lib/components/po-badge/po-badge.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoBadgeComponent } from './po-badge.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-badge.
 */
export class PoBadgeModule {
}
PoBadgeModule.ɵfac = function PoBadgeModule_Factory(t) { return new (t || PoBadgeModule)(); };
PoBadgeModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoBadgeModule });
PoBadgeModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoBadgeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [PoBadgeComponent],
                exports: [PoBadgeComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoBadgeModule, { declarations: [PoBadgeComponent], imports: [CommonModule], exports: [PoBadgeComponent] }); })();
