// @po-ui/ng-components/esm2020/lib/components/po-container/po-container.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoContainerComponent } from './po-container.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-container`.
 */
export class PoContainerModule {
}
PoContainerModule.ɵfac = function PoContainerModule_Factory(t) { return new (t || PoContainerModule)(); };
PoContainerModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoContainerModule });
PoContainerModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoContainerModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [PoContainerComponent],
                declarations: [PoContainerComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoContainerModule, { declarations: [PoContainerComponent], imports: [CommonModule], exports: [PoContainerComponent] }); })();
