// @po-ui/ng-templates/esm2020/lib/components/po-page-blocked-user/po-page-blocked-user.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoModule } from '@po-ui/ng-components';
import { PoPageBackgroundModule } from '../po-page-background/index';
import { PoPageBlockedUserComponent } from './po-page-blocked-user.component';
import { PoPageBlockedUserContactsComponent } from './po-page-blocked-user-contacts/po-page-blocked-user-contacts.component';
import { PoPageBlockedUserReasonComponent } from './po-page-blocked-user-reason/po-page-blocked-user-reason.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-page-blocked-user.
 */
export class PoPageBlockedUserModule {
}
PoPageBlockedUserModule.ɵfac = function PoPageBlockedUserModule_Factory(t) { return new (t || PoPageBlockedUserModule)(); };
PoPageBlockedUserModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageBlockedUserModule });
PoPageBlockedUserModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule, RouterModule, PoModule, PoPageBackgroundModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageBlockedUserModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule, PoModule, PoPageBackgroundModule],
                declarations: [PoPageBlockedUserComponent, PoPageBlockedUserContactsComponent, PoPageBlockedUserReasonComponent],
                exports: [PoPageBlockedUserComponent, PoPageBlockedUserContactsComponent, PoPageBlockedUserReasonComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageBlockedUserModule, { declarations: [PoPageBlockedUserComponent, PoPageBlockedUserContactsComponent, PoPageBlockedUserReasonComponent], imports: [CommonModule, RouterModule, PoModule, PoPageBackgroundModule], exports: [PoPageBlockedUserComponent, PoPageBlockedUserContactsComponent, PoPageBlockedUserReasonComponent] }); })();
