// @po-ui/ng-templates/esm2020/lib/components/po-modal-password-recovery/po-modal-password-recovery.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { PoFieldModule, PoI18nPipe, PoModalModule } from '@po-ui/ng-components';
import { PoModalPasswordRecoveryComponent } from './po-modal-password-recovery.component';
import { PoModalPasswordRecoveryErrorMessageComponent } from './po-modal-password-recovery-error-message/po-modal-password-recovery-error-message.component';
import { PoModalPasswordRecoveryService } from './po-modal-password-recovery.service';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do template do po-modal-password-recovery.
 */
export class PoModalPasswordRecoveryModule {
}
PoModalPasswordRecoveryModule.ɵfac = function PoModalPasswordRecoveryModule_Factory(t) { return new (t || PoModalPasswordRecoveryModule)(); };
PoModalPasswordRecoveryModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoModalPasswordRecoveryModule });
PoModalPasswordRecoveryModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [PoI18nPipe, PoModalPasswordRecoveryService], imports: [[CommonModule, FormsModule, PoFieldModule, PoModalModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoModalPasswordRecoveryModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, PoFieldModule, PoModalModule],
                declarations: [PoModalPasswordRecoveryComponent, PoModalPasswordRecoveryErrorMessageComponent],
                exports: [PoModalPasswordRecoveryComponent, PoModalPasswordRecoveryErrorMessageComponent],
                providers: [PoI18nPipe, PoModalPasswordRecoveryService]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoModalPasswordRecoveryModule, { declarations: [PoModalPasswordRecoveryComponent, PoModalPasswordRecoveryErrorMessageComponent], imports: [CommonModule, FormsModule, PoFieldModule, PoModalModule], exports: [PoModalPasswordRecoveryComponent, PoModalPasswordRecoveryErrorMessageComponent] }); })();
