// @po-ui/ng-components/esm2020/lib/components/po-progress/po-progress.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoProgressBarComponent } from './po-progress-bar/po-progress-bar.component';
import { PoProgressComponent } from './po-progress.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-progress`.
 */
export class PoProgressModule {
}
PoProgressModule.ɵfac = function PoProgressModule_Factory(t) { return new (t || PoProgressModule)(); };
PoProgressModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoProgressModule });
PoProgressModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoProgressModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [PoProgressComponent],
                declarations: [PoProgressBarComponent, PoProgressComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoProgressModule, { declarations: [PoProgressBarComponent, PoProgressComponent], imports: [CommonModule], exports: [PoProgressComponent] }); })();
